<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN">
<html>
<head>
<title>org.junit.tests.assertion.MultipleFailureExceptionTest.assertEmptyRethrowsSingleRuntimeException():v.45a2754f89c8ea8a5d2b4deeb88b1440_node7</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8">
<style type="text/css">
body { background-color:#fff; color:#000; font-family:Helvetica, sans-serif; font-size:10pt }
h1 { font-size:14pt }
.code { border-collapse:collapse; width:100%; font-family: "Andale Mono", monospace; font-size:10pt; line-height: 1.2em }
.color_black { color: black }
.color_blue { color: blue }
.color_green { color: green }
.color_orange { color: orange }
.color_red { color: red }
.line { border-left: 3px solid #ccc; padding-left: 1ex; white-space: pre; }
.msg { background-color:#fff8b4; color:#000; float:left; font-family:Helvetica, sans-serif; font-size: smaller; font-weight: bold; margin-bottom:10px; margin-top:10px; max-width:60em; padding:0.5em 1ex 0.5em 1ex; -webkit-border-radius:5px; -webkit-box-shadow:1px 1px 7px #000; white-space: pre-wrap; word-wrap: break-word; }
.num { background-color:#eee; color:#444; font-size: smaller; padding-right:2ex; text-align:right; width:2.5em; }
.dangling { background-color:white; color: gray; }
.visited { background-color:LemonChiffon; color: darkmagenta; }
.tooltip { display: none; background-color:#FFF0F0; border: 2px solid #F00; font-weight: normal; left:10em; padding: 2px; position: absolute; top: -1em; -webkit-border-radius:5px; -webkit-box-shadow:1px 1px 7px #000; z-index: 1}
.with_tooltip { position: relative; }
.with_tooltip:hover .tooltip, .visited:hover .tooltip { display: block; }
</style>
</head>
<body>
<center><h1>Cfg Node <a href="./../../MultipleFailureExceptionTest.java.5c3e6d8739a7da2006aef2128f5f1965.html#LINE42">7</a></h1></center>PROC: <a href="./../org.junit.tests.assertion.MultipleFailureExceptionTest.assertEmptyRethrowsSingleRuntimeException%28%29%3Av.45a2754f89c8ea8a5d2b4deeb88b1440.html">void MultipleFailureExceptionTest.assertEmptyRethrowsSingleRuntimeException()</a> LINE: <a href="./../../MultipleFailureExceptionTest.java.5c3e6d8739a7da2006aef2128f5f1965.html#LINE42">42</a>
<br>PREDS:
<a href="./../nodes/org.junit.tests.assertion.MultipleFailureExceptionTest.assertEmptyRethrowsSingleRuntimeException%28%29%3Av.45a2754f89c8ea8a5d2b4deeb88b1440_node6.html"><span class='visited with_tooltip'>N_6<span class='tooltip'>node6 preds:5 succs:7 exn:13 </span></span></a><br>SUCCS:
<a href="./../nodes/org.junit.tests.assertion.MultipleFailureExceptionTest.assertEmptyRethrowsSingleRuntimeException%28%29%3Av.45a2754f89c8ea8a5d2b4deeb88b1440_node10.html"><span class='visited with_tooltip'>N_10<span class='tooltip'>node10 preds:9 7 succs:2 exn:3 </span></span></a><br>EXN:
<a href="./../nodes/org.junit.tests.assertion.MultipleFailureExceptionTest.assertEmptyRethrowsSingleRuntimeException%28%29%3Av.45a2754f89c8ea8a5d2b4deeb88b1440_node13.html"><span class='visited with_tooltip'>N_13<span class='tooltip'>node13 preds: succs:11 12 exn:3 </span></span></a><br>
<LISTING class='color_green'>n$6=_fun_void Assert.fail()() [line 42];
</LISTING>
<hr width="100%">
<a name="session16" href="./../nodes/org.junit.tests.assertion.MultipleFailureExceptionTest.assertEmptyRethrowsSingleRuntimeException%28%29%3Av.45a2754f89c8ea8a5d2b4deeb88b1440_node7.html#session16">node7#session16</a>(<a href="./../../MultipleFailureExceptionTest.java.5c3e6d8739a7da2006aef2128f5f1965.html#LINE42">42</a>) [compute pre] liveness 7
<LISTING class='color_black'>JOIN

LEFT:   normal:{ }, exn:{ }
RIGHT:  normal:{ exception, return }, exn:{ }
RESULT: normal:{ exception, return }, exn:{ }

</LISTING>
<hr width="100%">
<a name="session17" href="./../nodes/org.junit.tests.assertion.MultipleFailureExceptionTest.assertEmptyRethrowsSingleRuntimeException%28%29%3Av.45a2754f89c8ea8a5d2b4deeb88b1440_node7.html#session17">node7#session17</a>(<a href="./../../MultipleFailureExceptionTest.java.5c3e6d8739a7da2006aef2128f5f1965.html#LINE42">42</a>) [exec] liveness 7
<LISTING class='color_black'>PRE STATE:
normal:{ exception, return }, exn:{ }

INSTR=  n$6=_fun_void Assert.fail()() [line 42]; 

STATE:
normal:{ exception, return }, exn:{ }

</LISTING>
<hr width="100%">
<a name="session33" href="./../nodes/org.junit.tests.assertion.MultipleFailureExceptionTest.assertEmptyRethrowsSingleRuntimeException%28%29%3Av.45a2754f89c8ea8a5d2b4deeb88b1440_node7.html#session33">node7#session33</a>(<a href="./../../MultipleFailureExceptionTest.java.5c3e6d8739a7da2006aef2128f5f1965.html#LINE42">42</a>) [compute pre] nullify
<LISTING class='color_black'></LISTING>
<hr width="100%">
<a name="session34" href="./../nodes/org.junit.tests.assertion.MultipleFailureExceptionTest.assertEmptyRethrowsSingleRuntimeException%28%29%3Av.45a2754f89c8ea8a5d2b4deeb88b1440_node7.html#session34">node7#session34</a>(<a href="./../../MultipleFailureExceptionTest.java.5c3e6d8739a7da2006aef2128f5f1965.html#LINE42">42</a>) [exec] nullify
<LISTING class='color_black'>PRE STATE:
(normal:{ exception }, exn:{ },normal:{ n$4, n$5, errors }, exn:{ })

INSTR=  n$6=_fun_void Assert.fail()() [line 42]; 

STATE:
(normal:{ exception }, exn:{ },normal:{ n$6 }, exn:{ })

</LISTING>
<hr width="100%">
<a name="session58" href="./../nodes/org.junit.tests.assertion.MultipleFailureExceptionTest.assertEmptyRethrowsSingleRuntimeException%28%29%3Av.45a2754f89c8ea8a5d2b4deeb88b1440_node7.html#session58">node7#session58</a>(<a href="./../../MultipleFailureExceptionTest.java.5c3e6d8739a7da2006aef2128f5f1965.html#LINE42">42</a>) [compute pre] devirtualizer analysis 7
<LISTING class='color_black'></LISTING>
<hr width="100%">
<a name="session59" href="./../nodes/org.junit.tests.assertion.MultipleFailureExceptionTest.assertEmptyRethrowsSingleRuntimeException%28%29%3Av.45a2754f89c8ea8a5d2b4deeb88b1440_node7.html#session59">node7#session59</a>(<a href="./../../MultipleFailureExceptionTest.java.5c3e6d8739a7da2006aef2128f5f1965.html#LINE42">42</a>) [exec] devirtualizer analysis 7
<LISTING class='color_black'>PRE STATE:
{ n$0 -&gt; org.junit.tests.assertion.MultipleFailureExceptionTest$ExpectedException,
  n$2 -&gt; org.junit.tests.assertion.MultipleFailureExceptionTest$ExpectedException,
  exception -&gt; org.junit.tests.assertion.MultipleFailureExceptionTest$ExpectedException }

INSTR=  n$6=_fun_void Assert.fail()() [line 42]; 

STATE UNCHANGED

INSTR=  EXIT_SCOPE(n$6); [line 42]; 

STATE UNCHANGED

INSTR=  APPLY_ABSTRACTION; [line 42]; 

STATE UNCHANGED

</LISTING>
<hr width="100%">
<a name="session91" href="./../nodes/org.junit.tests.assertion.MultipleFailureExceptionTest.assertEmptyRethrowsSingleRuntimeException%28%29%3Av.45a2754f89c8ea8a5d2b4deeb88b1440_node7.html#session91">node7#session91</a>(<a href="./../../MultipleFailureExceptionTest.java.5c3e6d8739a7da2006aef2128f5f1965.html#LINE42">42</a>) [exec] devirtualizer
<LISTING class='color_black'></LISTING>
<hr width="100%">
<a name="session92" href="./../nodes/org.junit.tests.assertion.MultipleFailureExceptionTest.assertEmptyRethrowsSingleRuntimeException%28%29%3Av.45a2754f89c8ea8a5d2b4deeb88b1440_node7.html#session92">node7#session92</a>(<a href="./../../MultipleFailureExceptionTest.java.5c3e6d8739a7da2006aef2128f5f1965.html#LINE42">42</a>) [exec] devirtualizer
<LISTING class='color_black'></LISTING>
<hr width="100%">
<a name="session93" href="./../nodes/org.junit.tests.assertion.MultipleFailureExceptionTest.assertEmptyRethrowsSingleRuntimeException%28%29%3Av.45a2754f89c8ea8a5d2b4deeb88b1440_node7.html#session93">node7#session93</a>(<a href="./../../MultipleFailureExceptionTest.java.5c3e6d8739a7da2006aef2128f5f1965.html#LINE42">42</a>) [exec] devirtualizer
<LISTING class='color_black'></LISTING>
<hr width="100%">
<a name="session8" href="./../nodes/org.junit.tests.assertion.MultipleFailureExceptionTest.assertEmptyRethrowsSingleRuntimeException%28%29%3Av.45a2754f89c8ea8a5d2b4deeb88b1440_node7.html#session8">node7#session8</a>(<a href="./../../MultipleFailureExceptionTest.java.5c3e6d8739a7da2006aef2128f5f1965.html#LINE42">42</a>) [compute pre] starvation
<LISTING class='color_black'></LISTING>
<hr width="100%">
<a name="session9" href="./../nodes/org.junit.tests.assertion.MultipleFailureExceptionTest.assertEmptyRethrowsSingleRuntimeException%28%29%3Av.45a2754f89c8ea8a5d2b4deeb88b1440_node7.html#session9">node7#session9</a>(<a href="./../../MultipleFailureExceptionTest.java.5c3e6d8739a7da2006aef2128f5f1965.html#LINE42">42</a>) [exec] starvation
<LISTING class='color_black'>PRE STATE:
{guard_map= { }; lock_state= {map= { }; acquisitions= { }};
 critical_pairs= { }; attributes= { }; thread= UnknownThread;
 scheduled_work= { }; var_state= { }}

INSTR=  n$6=_fun_void Assert.fail()() [line 42]; 

STATE:
{guard_map= { }; lock_state= {map= { }; acquisitions= { }};
 critical_pairs= { }; attributes= { }; thread= UnknownThread;
 scheduled_work= { }; var_state= { }}

INSTR=  EXIT_SCOPE(n$6); [line 42]; 

STATE:
{guard_map= { }; lock_state= {map= { }; acquisitions= { }};
 critical_pairs= { }; attributes= { }; thread= UnknownThread;
 scheduled_work= { }; var_state= { }}

INSTR=  APPLY_ABSTRACTION; [line 42]; 

STATE UNCHANGED

</LISTING>
<hr width="100%">
<a name="session21" href="./../nodes/org.junit.tests.assertion.MultipleFailureExceptionTest.assertEmptyRethrowsSingleRuntimeException%28%29%3Av.45a2754f89c8ea8a5d2b4deeb88b1440_node7.html#session21">node7#session21</a>(<a href="./../../MultipleFailureExceptionTest.java.5c3e6d8739a7da2006aef2128f5f1965.html#LINE42">42</a>) [compute pre] racerd
<LISTING class='color_black'></LISTING>
<hr width="100%">
<a name="session22" href="./../nodes/org.junit.tests.assertion.MultipleFailureExceptionTest.assertEmptyRethrowsSingleRuntimeException%28%29%3Av.45a2754f89c8ea8a5d2b4deeb88b1440_node7.html#session22">node7#session22</a>(<a href="./../../MultipleFailureExceptionTest.java.5c3e6d8739a7da2006aef2128f5f1965.html#LINE42">42</a>) [exec] racerd
<LISTING class='color_black'>PRE STATE:
Threads: NoThread, Locks: 0 
Accesses { } 
Ownership: { n$0 -&gt; Owned,
             n$1 -&gt; Unowned,
             n$3 -&gt; Owned,
             n$5 -&gt; Owned,
             exception -&gt; Owned,
             errors -&gt; Owned,
             this -&gt; OwnedIf{ 0 } } 
Attributes: { } 


INSTR=  n$6 := void Assert.fail()({ }) [line 42]; 

STATE:
Threads: NoThread, Locks: 0 
Accesses { } 
Ownership: { n$0 -&gt; Owned,
             n$1 -&gt; Unowned,
             n$3 -&gt; Owned,
             n$5 -&gt; Owned,
             n$6 -&gt; Unowned,
             exception -&gt; Owned,
             errors -&gt; Owned,
             this -&gt; OwnedIf{ 0 } } 
Attributes: { } 


INSTR=  EXIT_SCOPE(n$6); [line 42]; 

STATE UNCHANGED

INSTR=  APPLY_ABSTRACTION; [line 42]; 

STATE UNCHANGED

</LISTING>
<hr width="100%">
<a name="session36" href="./../nodes/org.junit.tests.assertion.MultipleFailureExceptionTest.assertEmptyRethrowsSingleRuntimeException%28%29%3Av.45a2754f89c8ea8a5d2b4deeb88b1440_node7.html#session36">node7#session36</a>(<a href="./../../MultipleFailureExceptionTest.java.5c3e6d8739a7da2006aef2128f5f1965.html#LINE42">42</a>) biabduction
<LISTING class='color_black'>**** [FP:Pending] void MultipleFailureExceptionTest.assertEmptyRethrowsSingleRuntimeException() Node: 7, Procedure: void MultipleFailureExceptionTest.assertEmptyRethrowsSingleRuntimeException(), Todo: 2 ****
  PROP 1 of 2:
  <span class='color_blue'><span class='color_red'>(@f$2  &lt;=  -1)</span>; <span class='color_red'>MEMne &lt; new:38 &gt; <span class='color_orange'>{vpath: }</span>(exception)</span>; <span class='color_red'>UND &lt; List Collections.singletonList(MultipleFailureExceptionTest$ExpectedException) &gt; :39(@f$0)</span> ;
  <span class='color_red'>exception = _t$0<span class='color_orange'>update:38</span>:java.lang.Throwable*</span>; <span class='color_red'>CatchVar25 = _t$1<span class='color_orange'>initial</span>:void</span>; <span class='color_red'>return = _t$2<span class='color_orange'>initial</span>:void</span>; <span class='color_red'>errors = 0<span class='color_orange'>nullify</span>:java.util.List*</span>; <span class='color_red'>this = @f$1<span class='color_orange'>formal</span>:org.junit.tests.assertion.MultipleFailureExceptionTest*( sub )</span>; <span class='color_red'>old_this = @f$1<span class='color_orange'>formal</span>:org.junit.tests.assertion.MultipleFailureExceptionTest*( sub )</span>; <span class='color_red'>e = _t$3<span class='color_orange'>initial</span>:org.junit.tests.assertion.MultipleFailureExceptionTest$ExpectedException*</span> ;
  <span class='color_red'>exception|->{}<span class='color_orange'>alloc</span>:org.junit.tests.assertion.MultipleFailureExceptionTest$ExpectedException</span>; <span class='color_red'>@f$0|->{mIsEmpty:1<span class='color_orange'>update:(z)41</span>}<span class='color_orange'>formal(z)</span>:java.util.List( sub )</span>
  [footprint
     <span class='color_red'>(@f$2  &lt;=  -1)</span> *
     <span class='color_red'>$RET_java.util.Collections.singletonList(org.junit.tests.assertion.MultipleFailureExceptionTest$ExpectedException):java.util.List|abducedRetvar = @f$0<span class='color_orange'>none</span>:java.util.List*( sub )</span>; <span class='color_red'>this = @f$1<span class='color_orange'>formal</span>:org.junit.tests.assertion.MultipleFailureExceptionTest*( sub )</span> ;
     <span class='color_red'>@f$0|->{mIsEmpty:@f$2<span class='color_orange'>rearrange:(z)27</span>}<span class='color_orange'>rearrange:(z)88</span>:java.util.List( sub )</span>  ]</span>
  PROP 2 of 2:
  <span class='color_blue'><span class='color_red'>@f$2 = 1</span> ;
  <span class='color_red'>MEMne &lt; new:38 &gt; <span class='color_orange'>{vpath: }</span>(exception)</span>; <span class='color_red'>UND &lt; List Collections.singletonList(MultipleFailureExceptionTest$ExpectedException) &gt; :39(@f$0)</span> ;
  <span class='color_red'>exception = _t$0<span class='color_orange'>update:38</span>:java.lang.Throwable*</span>; <span class='color_red'>CatchVar25 = _t$1<span class='color_orange'>initial</span>:void</span>; <span class='color_red'>return = _t$2<span class='color_orange'>initial</span>:void</span>; <span class='color_red'>errors = 0<span class='color_orange'>nullify</span>:java.util.List*</span>; <span class='color_red'>this = @f$1<span class='color_orange'>formal</span>:org.junit.tests.assertion.MultipleFailureExceptionTest*( sub )</span>; <span class='color_red'>old_this = @f$1<span class='color_orange'>formal</span>:org.junit.tests.assertion.MultipleFailureExceptionTest*( sub )</span>; <span class='color_red'>e = _t$3<span class='color_orange'>initial</span>:org.junit.tests.assertion.MultipleFailureExceptionTest$ExpectedException*</span> ;
  <span class='color_red'>exception|->{}<span class='color_orange'>alloc</span>:org.junit.tests.assertion.MultipleFailureExceptionTest$ExpectedException</span>; <span class='color_red'>@f$0|->{mIsEmpty:1<span class='color_orange'>formal(z)</span>}<span class='color_orange'>formal(z)</span>:java.util.List( sub )</span>
  [footprint
     <span class='color_red'>@f$2 = 1</span> *
     <span class='color_red'>$RET_java.util.Collections.singletonList(org.junit.tests.assertion.MultipleFailureExceptionTest$ExpectedException):java.util.List|abducedRetvar = @f$0<span class='color_orange'>none</span>:java.util.List*( sub )</span>; <span class='color_red'>this = @f$1<span class='color_orange'>formal</span>:org.junit.tests.assertion.MultipleFailureExceptionTest*( sub )</span> ;
     <span class='color_red'>@f$0|->{mIsEmpty:1<span class='color_orange'>rearrange:(z)27</span>}<span class='color_orange'>rearrange:(z)88</span>:java.util.List( sub )</span>  ]</span>
  .... Instructions: ....
  <span class='color_green'>n$6=_fun_void Assert.fail()() [line 42];
  EXIT_SCOPE(n$6); [line 42];
  APPLY_ABSTRACTION; [line 42];
  </span>
  
  Processing prop 1/2
    Instruction <span class='color_green'>n$6=_fun_void Assert.fail()() [line 42]</span>
    in vpath_find exp:n$196
    vpath_find: found exception :java.lang.Throwable*
    Found 1 specs for function org.junit.Assert.fail():void
    START EXECUTING SPECS FOR org.junit.Assert.fail():void from state
    <span class='color_blue'>(@f$2  &lt;=  -1); MEMne &lt; new:38 &gt; <span class='color_orange'>{vpath: }</span>(exception); UND &lt; List Collections.singletonList(MultipleFailureExceptionTest$ExpectedException) &gt; :39(@f$0) ;
    exception = n$196<span class='color_orange'>update:38</span>:java.lang.Throwable*; CatchVar25 = n$197<span class='color_orange'>initial</span>:void; return = n$198<span class='color_orange'>initial</span>:void; errors = 0<span class='color_orange'>nullify</span>:java.util.List*; this = @f$1<span class='color_orange'>formal</span>:org.junit.tests.assertion.MultipleFailureExceptionTest*( sub ); old_this = @f$1<span class='color_orange'>formal</span>:org.junit.tests.assertion.MultipleFailureExceptionTest*( sub ); e = n$199<span class='color_orange'>initial</span>:org.junit.tests.assertion.MultipleFailureExceptionTest$ExpectedException* ;
    exception|->{}<span class='color_orange'>alloc</span>:org.junit.tests.assertion.MultipleFailureExceptionTest$ExpectedException; @f$0|->{mIsEmpty:1<span class='color_orange'>update:(z)41</span>}<span class='color_orange'>formal(z)</span>:java.util.List( sub )
    [footprint
       (@f$2  &lt;=  -1) *
       $RET_java.util.Collections.singletonList(org.junit.tests.assertion.MultipleFailureExceptionTest$ExpectedException):java.util.List|abducedRetvar = @f$0<span class='color_orange'>none</span>:java.util.List*( sub ); this = @f$1<span class='color_orange'>formal</span>:org.junit.tests.assertion.MultipleFailureExceptionTest*( sub ) ;
       @f$0|->{mIsEmpty:@f$2<span class='color_orange'>rearrange:(z)27</span>}<span class='color_orange'>rearrange:(z)88</span>:java.util.List( sub )  ]</span>
    EXECUTING SPEC 1/1
    ACTUAL PRECONDITION =
      <span class='color_blue'>(@f$2  &lt;=  -1); MEMne &lt; new:38 &gt; <span class='color_orange'>{vpath: }</span>(exception); UND &lt; List Collections.singletonList(MultipleFailureExceptionTest$ExpectedException) &gt; :39(@f$0) ;
      exception = n$196<span class='color_orange'>update:38</span>:java.lang.Throwable*; CatchVar25 = n$197<span class='color_orange'>initial</span>:void; return = n$198<span class='color_orange'>initial</span>:void; errors = 0<span class='color_orange'>nullify</span>:java.util.List*; this = @f$1<span class='color_orange'>formal</span>:org.junit.tests.assertion.MultipleFailureExceptionTest*( sub ); old_this = @f$1<span class='color_orange'>formal</span>:org.junit.tests.assertion.MultipleFailureExceptionTest*( sub ); e = n$199<span class='color_orange'>initial</span>:org.junit.tests.assertion.MultipleFailureExceptionTest$ExpectedException* ;
      exception|->{}<span class='color_orange'>alloc</span>:org.junit.tests.assertion.MultipleFailureExceptionTest$ExpectedException; @f$0|->{mIsEmpty:1<span class='color_orange'>update:(z)41</span>}<span class='color_orange'>formal(z)</span>:java.util.List( sub )
      [footprint
         (@f$2  &lt;=  -1) *
         $RET_java.util.Collections.singletonList(org.junit.tests.assertion.MultipleFailureExceptionTest$ExpectedException):java.util.List|abducedRetvar = @f$0<span class='color_orange'>none</span>:java.util.List*( sub ); this = @f$1<span class='color_orange'>formal</span>:org.junit.tests.assertion.MultipleFailureExceptionTest*( sub ) ;
         @f$0|->{mIsEmpty:@f$2<span class='color_orange'>rearrange:(z)27</span>}<span class='color_orange'>rearrange:(z)88</span>:java.util.List( sub )  ]</span>
    SPEC =
      --------------------------- ---------- ---------------------------
      PRE:
      <span class='color_blue'><span class='color_blue'></span></span>
      POST 1 of 1:
      <span class='color_blue'><span class='color_red'>MEMne &lt; void Assert.fail(String):96 &gt; <span class='color_orange'>{vpath: }</span>(_t$292)</span> ;
      <span class='color_red'>return|callee = EXN _t$292<span class='color_orange'>update:96</span>:void</span> ;
      <span class='color_red'>_t$292|->{}<span class='color_orange'>alloc</span>:java.lang.AssertionError</span></span>
      ----------------------------------------------------------------
    pre_check_pure_implication
    pi1:
      (@f$2  &lt;=  -1); MEMne &lt; new:38 &gt; <span class='color_orange'>{vpath: }</span>(n$196); UND &lt; List Collections.singletonList(MultipleFailureExceptionTest$ExpectedException) &gt; :39(@f$0)
    pi2:
      
    returns
    sub1:
      
    sub2:
      
    Final Implication
    SUB:
      
    <span class='color_blue'>(@f$2  &lt;=  -1); MEMne &lt; new:38 &gt; <span class='color_orange'>{vpath: }</span>(exception); UND &lt; List Collections.singletonList(MultipleFailureExceptionTest$ExpectedException) &gt; :39(@f$0) ;
    exception = n$196<span class='color_orange'>update:38</span>:java.lang.Throwable*; CatchVar25 = n$197<span class='color_orange'>initial</span>:void; return = n$198<span class='color_orange'>initial</span>:void; errors = 0<span class='color_orange'>nullify</span>:java.util.List*; this = @f$1<span class='color_orange'>formal</span>:org.junit.tests.assertion.MultipleFailureExceptionTest*( sub ); old_this = @f$1<span class='color_orange'>formal</span>:org.junit.tests.assertion.MultipleFailureExceptionTest*( sub ); e = n$199<span class='color_orange'>initial</span>:org.junit.tests.assertion.MultipleFailureExceptionTest$ExpectedException* ;
    exception|->{}<span class='color_orange'>alloc</span>:org.junit.tests.assertion.MultipleFailureExceptionTest$ExpectedException; @f$0|->{mIsEmpty:1<span class='color_orange'>update:(z)41</span>}<span class='color_orange'>formal(z)</span>:java.util.List( sub )
    [footprint
       (@f$2  &lt;=  -1) *
       $RET_java.util.Collections.singletonList(org.junit.tests.assertion.MultipleFailureExceptionTest$ExpectedException):java.util.List|abducedRetvar = @f$0<span class='color_orange'>none</span>:java.util.List*( sub ); this = @f$1<span class='color_orange'>formal</span>:org.junit.tests.assertion.MultipleFailureExceptionTest*( sub ) ;
       @f$0|->{mIsEmpty:@f$2<span class='color_orange'>rearrange:(z)27</span>}<span class='color_orange'>rearrange:(z)88</span>:java.util.List( sub )  ]</span>
    |-
    <span class='color_blue'></span>
    Result of Abduction
      SUB:
        
      <span class='color_blue'>(@f$2  &lt;=  -1); MEMne &lt; new:38 &gt; <span class='color_orange'>{vpath: }</span>(exception); UND &lt; List Collections.singletonList(MultipleFailureExceptionTest$ExpectedException) &gt; :39(@f$0) ;
      exception = n$196<span class='color_orange'>update:38</span>:java.lang.Throwable*; CatchVar25 = n$197<span class='color_orange'>initial</span>:void; return = n$198<span class='color_orange'>initial</span>:void; errors = 0<span class='color_orange'>nullify</span>:java.util.List*; this = @f$1<span class='color_orange'>formal</span>:org.junit.tests.assertion.MultipleFailureExceptionTest*( sub ); old_this = @f$1<span class='color_orange'>formal</span>:org.junit.tests.assertion.MultipleFailureExceptionTest*( sub ); e = n$199<span class='color_orange'>initial</span>:org.junit.tests.assertion.MultipleFailureExceptionTest$ExpectedException* ;
      exception|->{}<span class='color_orange'>alloc</span>:org.junit.tests.assertion.MultipleFailureExceptionTest$ExpectedException; @f$0|->{mIsEmpty:1<span class='color_orange'>update:(z)41</span>}<span class='color_orange'>formal(z)</span>:java.util.List( sub )
      [footprint
         (@f$2  &lt;=  -1) *
         $RET_java.util.Collections.singletonList(org.junit.tests.assertion.MultipleFailureExceptionTest$ExpectedException):java.util.List|abducedRetvar = @f$0<span class='color_orange'>none</span>:java.util.List*( sub ); this = @f$1<span class='color_orange'>formal</span>:org.junit.tests.assertion.MultipleFailureExceptionTest*( sub ) ;
         @f$0|->{mIsEmpty:@f$2<span class='color_orange'>rearrange:(z)27</span>}<span class='color_orange'>rearrange:(z)88</span>:java.util.List( sub )  ]</span>
      |-
      <span class='color_blue'></span>
    returning TRUE
    in vpath_find exp:_t$292
    vpath_find: cannot find _t$292
      New footprint:
      
      Frame fld:
      
      Missing fld:
      
      Instantiated frame:
      n$196|->{}<span class='color_orange'>alloc</span>:org.junit.tests.assertion.MultipleFailureExceptionTest$ExpectedException; @f$0|->{mIsEmpty:1<span class='color_orange'>update:(z)41</span>}<span class='color_orange'>formal(z)</span>:java.util.List( sub ); &amp;exception|->n$196<span class='color_orange'>update:38</span>:java.lang.Throwable*; &amp;CatchVar25|->n$197<span class='color_orange'>initial</span>:void; &amp;return|->n$198<span class='color_orange'>initial</span>:void; &amp;errors|->0<span class='color_orange'>nullify</span>:java.util.List*; &amp;this|->@f$1<span class='color_orange'>formal</span>:org.junit.tests.assertion.MultipleFailureExceptionTest*( sub ); &amp;old_this|->@f$1<span class='color_orange'>formal</span>:org.junit.tests.assertion.MultipleFailureExceptionTest*( sub ); &amp;e|->n$199<span class='color_orange'>initial</span>:org.junit.tests.assertion.MultipleFailureExceptionTest$ExpectedException*
      Instantiated post:
      PROP 1 of 1:
      <span class='color_blue'><span class='color_red'>MEMne &lt; void Assert.fail():42 &gt; <span class='color_orange'>{vpath: }</span>(_t$292)</span> ;
      <span class='color_red'>return|callee = EXN _t$292<span class='color_orange'>update:42</span>:void</span> ;
      <span class='color_red'>_t$292|->{}<span class='color_orange'>alloc</span>:java.lang.AssertionError</span></span>
      
    ***** RESULTS FUNCTION CALL *******
    PROP 1 of 1:
    <span class='color_blue'><span class='color_blue'>(@f$2  &lt;=  -1)</span>; <span class='color_red'>MEMne &lt; void Assert.fail():42 &gt; <span class='color_orange'>{vpath: }</span>(_t$0)</span>; <span class='color_blue'>MEMne &lt; new:38 &gt; <span class='color_orange'>{vpath: }</span>(exception)</span>; <span class='color_blue'>UND &lt; List Collections.singletonList(MultipleFailureExceptionTest$ExpectedException) &gt; :39(@f$0)</span> ;
    <span class='color_blue'>exception = n$196<span class='color_orange'>update:38</span>:java.lang.Throwable*</span>; <span class='color_blue'>CatchVar25 = n$197<span class='color_orange'>initial</span>:void</span>; <span class='color_blue'>return = <span class='color_red'>EXN _t$0<span class='color_orange'>update:42</span></span>:void</span>; <span class='color_blue'>errors = 0<span class='color_orange'>nullify</span>:java.util.List*</span>; <span class='color_blue'>this = @f$1<span class='color_orange'>formal</span>:org.junit.tests.assertion.MultipleFailureExceptionTest*( sub )</span>; <span class='color_blue'>old_this = @f$1<span class='color_orange'>formal</span>:org.junit.tests.assertion.MultipleFailureExceptionTest*( sub )</span>; <span class='color_blue'>e = n$199<span class='color_orange'>initial</span>:org.junit.tests.assertion.MultipleFailureExceptionTest$ExpectedException*</span> ;
    <span class='color_red'>_t$0|->{}<span class='color_orange'>alloc</span>:java.lang.AssertionError</span>; <span class='color_blue'>exception|->{}<span class='color_orange'>alloc</span>:org.junit.tests.assertion.MultipleFailureExceptionTest$ExpectedException</span>; <span class='color_blue'>@f$0|->{mIsEmpty:1<span class='color_orange'>update:(z)41</span>}<span class='color_orange'>formal(z)</span>:java.util.List( sub )</span>
    [footprint
       <span class='color_blue'>(@f$2  &lt;=  -1)</span> *
       <span class='color_blue'>$RET_java.util.Collections.singletonList(org.junit.tests.assertion.MultipleFailureExceptionTest$ExpectedException):java.util.List|abducedRetvar = @f$0<span class='color_orange'>none</span>:java.util.List*( sub )</span>; <span class='color_blue'>this = @f$1<span class='color_orange'>formal</span>:org.junit.tests.assertion.MultipleFailureExceptionTest*( sub )</span> ;
       <span class='color_blue'>@f$0|->{mIsEmpty:@f$2<span class='color_orange'>rearrange:(z)27</span>}<span class='color_orange'>rearrange:(z)88</span>:java.util.List( sub )</span>  ]</span>
    ***** END RESULTS FUNCTION CALL *******
    explain_dereference_as_caller_expression @f$0: cannot explain None
    <span class='color_red'>found error in dereference</span>
    spec_pre:
    <span class='color_blue'>(@f$2  &lt;=  -1); MEMne &lt; void Assert.fail():42 &gt; <span class='color_orange'>{vpath: }</span>(_t$292); MEMne &lt; new:38 &gt; <span class='color_orange'>{vpath: }</span>(exception); UND &lt; List Collections.singletonList(MultipleFailureExceptionTest$ExpectedException) &gt; :39(@f$0) ;
    exception = n$196<span class='color_orange'>update:38</span>:java.lang.Throwable*; CatchVar25 = n$197<span class='color_orange'>initial</span>:void; return = EXN _t$292<span class='color_orange'>update:42</span>:void; errors = 0<span class='color_orange'>nullify</span>:java.util.List*; this = @f$1<span class='color_orange'>formal</span>:org.junit.tests.assertion.MultipleFailureExceptionTest*( sub ); old_this = @f$1<span class='color_orange'>formal</span>:org.junit.tests.assertion.MultipleFailureExceptionTest*( sub ); e = n$199<span class='color_orange'>initial</span>:org.junit.tests.assertion.MultipleFailureExceptionTest$ExpectedException* ;
    _t$292|->{}<span class='color_orange'>alloc</span>:java.lang.AssertionError; exception|->{}<span class='color_orange'>alloc</span>:org.junit.tests.assertion.MultipleFailureExceptionTest$ExpectedException; @f$0|->{mIsEmpty:1<span class='color_orange'>update:(z)41</span>}<span class='color_orange'>formal(z)</span>:java.util.List( sub )
    [footprint
       (@f$2  &lt;=  -1) *
       $RET_java.util.Collections.singletonList(org.junit.tests.assertion.MultipleFailureExceptionTest$ExpectedException):java.util.List|abducedRetvar = @f$0<span class='color_orange'>none</span>:java.util.List*( sub ); this = @f$1<span class='color_orange'>formal</span>:org.junit.tests.assertion.MultipleFailureExceptionTest*( sub ) ;
       @f$0|->{mIsEmpty:@f$2<span class='color_orange'>rearrange:(z)27</span>}<span class='color_orange'>rearrange:(z)88</span>:java.util.List( sub )  ]</span>
    exp @f$0 desc: 
    Prover.check_inconsistency_base: inconsistency reason [None]
    Prover.check_inconsistency_base: inconsistency reason [None]
    Prover.check_inconsistency_base: inconsistency reason [None]
    Prover.check_inconsistency_base: inconsistency reason [None]
    Instruction Returns
    PROP 1 of 1:
    <span class='color_blue'><span class='color_blue'>(@f$2  &lt;=  -1)</span>; <span class='color_red'>MEMne &lt; new:38 &gt; <span class='color_orange'>{vpath: }</span>(exception)</span>; <span class='color_blue'>MEMne &lt; void Assert.fail():42 &gt; <span class='color_orange'>{vpath: }</span>(_t$0)</span>; <span class='color_blue'>UND &lt; List Collections.singletonList(MultipleFailureExceptionTest$ExpectedException) &gt; :39(@f$0)</span>; <span class='color_red'>RET &lt; void Assert.fail() &gt; (n$6)</span> ;
    <span class='color_blue'>exception = <span class='color_red'>_t$1<span class='color_orange'>update:38</span></span>:java.lang.Throwable*</span>; <span class='color_blue'>CatchVar25 = <span class='color_red'>_t$2<span class='color_orange'>initial</span></span>:void</span>; <span class='color_blue'>return = <span class='color_red'>EXN _t$0<span class='color_orange'>update:42</span></span>:void</span>; <span class='color_blue'>errors = 0<span class='color_orange'>nullify</span>:java.util.List*</span>; <span class='color_blue'>this = @f$1<span class='color_orange'>formal</span>:org.junit.tests.assertion.MultipleFailureExceptionTest*( sub )</span>; <span class='color_blue'>old_this = @f$1<span class='color_orange'>formal</span>:org.junit.tests.assertion.MultipleFailureExceptionTest*( sub )</span>; <span class='color_blue'>e = _t$3<span class='color_orange'>initial</span>:org.junit.tests.assertion.MultipleFailureExceptionTest$ExpectedException*</span> ;
    <span class='color_red'>exception|->{}<span class='color_orange'>alloc</span>:org.junit.tests.assertion.MultipleFailureExceptionTest$ExpectedException</span>; <span class='color_blue'>_t$0|->{}<span class='color_orange'>alloc</span>:java.lang.AssertionError</span>; <span class='color_blue'>@f$0|->{mIsEmpty:1<span class='color_orange'>update:(z)41</span>}<span class='color_orange'>formal(z)</span>:java.util.List( sub )</span>
    [footprint
       <span class='color_blue'>(@f$2  &lt;=  -1)</span> *
       <span class='color_blue'>$RET_java.util.Collections.singletonList(org.junit.tests.assertion.MultipleFailureExceptionTest$ExpectedException):java.util.List|abducedRetvar = @f$0<span class='color_orange'>none</span>:java.util.List*( sub )</span>; <span class='color_blue'>this = @f$1<span class='color_orange'>formal</span>:org.junit.tests.assertion.MultipleFailureExceptionTest*( sub )</span> ;
       <span class='color_blue'>@f$0|->{mIsEmpty:@f$2<span class='color_orange'>rearrange:(z)27</span>}<span class='color_orange'>rearrange:(z)88</span>:java.util.List( sub )</span>  ]</span>
    
    Instruction <span class='color_green'>EXIT_SCOPE(n$6); [line 42]</span>
    in vpath_find exp:_t$1
    vpath_find: found exception :java.lang.Throwable*
    in vpath_find exp:_t$0
    vpath_find: cannot find _t$0
    Instruction Returns
    PROP 1 of 1:
    <span class='color_blue'><span class='color_blue'>(@f$2  &lt;=  -1)</span>; <span class='color_blue'>MEMne &lt; new:38 &gt; <span class='color_orange'>{vpath: }</span>(exception)</span>; <span class='color_blue'>MEMne &lt; void Assert.fail():42 &gt; <span class='color_orange'>{vpath: }</span>(_t$0)</span>; <span class='color_blue'>UND &lt; List Collections.singletonList(MultipleFailureExceptionTest$ExpectedException) &gt; :39(@f$0)</span>; <span class='color_red'>RET &lt; void Assert.fail() &gt; (_t$4)</span> ;
    <span class='color_blue'>exception = _t$1<span class='color_orange'>update:38</span>:java.lang.Throwable*</span>; <span class='color_blue'>CatchVar25 = _t$2<span class='color_orange'>initial</span>:void</span>; <span class='color_blue'>return = EXN _t$0<span class='color_orange'>update:42</span>:void</span>; <span class='color_blue'>errors = 0<span class='color_orange'>nullify</span>:java.util.List*</span>; <span class='color_blue'>this = @f$1<span class='color_orange'>formal</span>:org.junit.tests.assertion.MultipleFailureExceptionTest*( sub )</span>; <span class='color_blue'>old_this = @f$1<span class='color_orange'>formal</span>:org.junit.tests.assertion.MultipleFailureExceptionTest*( sub )</span>; <span class='color_blue'>e = _t$3<span class='color_orange'>initial</span>:org.junit.tests.assertion.MultipleFailureExceptionTest$ExpectedException*</span> ;
    <span class='color_blue'>exception|->{}<span class='color_orange'>alloc</span>:org.junit.tests.assertion.MultipleFailureExceptionTest$ExpectedException</span>; <span class='color_blue'>_t$0|->{}<span class='color_orange'>alloc</span>:java.lang.AssertionError</span>; <span class='color_blue'>@f$0|->{mIsEmpty:1<span class='color_orange'>update:(z)41</span>}<span class='color_orange'>formal(z)</span>:java.util.List( sub )</span>
    [footprint
       <span class='color_blue'>(@f$2  &lt;=  -1)</span> *
       <span class='color_blue'>$RET_java.util.Collections.singletonList(org.junit.tests.assertion.MultipleFailureExceptionTest$ExpectedException):java.util.List|abducedRetvar = @f$0<span class='color_orange'>none</span>:java.util.List*( sub )</span>; <span class='color_blue'>this = @f$1<span class='color_orange'>formal</span>:org.junit.tests.assertion.MultipleFailureExceptionTest*( sub )</span> ;
       <span class='color_blue'>@f$0|->{mIsEmpty:@f$2<span class='color_orange'>rearrange:(z)27</span>}<span class='color_orange'>rearrange:(z)88</span>:java.util.List( sub )</span>  ]</span>
    
    Instruction <span class='color_green'>APPLY_ABSTRACTION; [line 42]</span>
    in vpath_find exp:_t$1
    vpath_find: found exception :java.lang.Throwable*
    in vpath_find exp:_t$0
    vpath_find: cannot find _t$0
    Prover.check_inconsistency_base: inconsistency reason [None]
    Prover.check_inconsistency_base: inconsistency reason [None]
    Can't find field mIsEmpty in StrexpMatch.find
    Can't find field mIsEmpty in StrexpMatch.find
    <span class='color_blue'>(@f$2  &lt;=  -1); MEMne &lt; new:38 &gt; <span class='color_orange'>{vpath: }</span>(exception); MEMne &lt; void Assert.fail():42 &gt; <span class='color_orange'>{vpath: }</span>(_t$0); UND &lt; List Collections.singletonList(MultipleFailureExceptionTest$ExpectedException) &gt; :39(@f$0) ;
    exception = _t$1<span class='color_orange'>update:38</span>:java.lang.Throwable*; CatchVar25 = _t$2<span class='color_orange'>initial</span>:void; return = EXN _t$0<span class='color_orange'>update:42</span>:void; errors = 0<span class='color_orange'>nullify</span>:java.util.List*; this = @f$1<span class='color_orange'>formal</span>:org.junit.tests.assertion.MultipleFailureExceptionTest*( sub ); old_this = @f$1<span class='color_orange'>formal</span>:org.junit.tests.assertion.MultipleFailureExceptionTest*( sub ); e = _t$3<span class='color_orange'>initial</span>:org.junit.tests.assertion.MultipleFailureExceptionTest$ExpectedException* ;
    exception|->{}<span class='color_orange'>alloc</span>:org.junit.tests.assertion.MultipleFailureExceptionTest$ExpectedException; _t$0|->{}<span class='color_orange'>alloc</span>:java.lang.AssertionError; @f$0|->{mIsEmpty:1<span class='color_orange'>update:(z)41</span>}<span class='color_orange'>formal(z)</span>:java.util.List( sub )
    [footprint
       (@f$2  &lt;=  -1) *
       $RET_java.util.Collections.singletonList(org.junit.tests.assertion.MultipleFailureExceptionTest$ExpectedException):java.util.List|abducedRetvar = @f$0<span class='color_orange'>none</span>:java.util.List*( sub ); this = @f$1<span class='color_orange'>formal</span>:org.junit.tests.assertion.MultipleFailureExceptionTest*( sub ) ;
       @f$0|->{mIsEmpty:@f$2<span class='color_orange'>rearrange:(z)27</span>}<span class='color_orange'>rearrange:(z)88</span>:java.util.List( sub )  ]</span>
    Instruction Returns
    PROP 1 of 1:
    <span class='color_blue'><span class='color_blue'>(@f$2  &lt;=  -1)</span>; <span class='color_blue'>MEMne &lt; new:38 &gt; <span class='color_orange'>{vpath: }</span>(exception)</span>; <span class='color_blue'>MEMne &lt; void Assert.fail():42 &gt; <span class='color_orange'>{vpath: }</span>(_t$0)</span>; <span class='color_blue'>UND &lt; List Collections.singletonList(MultipleFailureExceptionTest$ExpectedException) &gt; :39(@f$0)</span> ;
    <span class='color_blue'>exception = _t$1<span class='color_orange'>update:38</span>:java.lang.Throwable*</span>; <span class='color_blue'>CatchVar25 = _t$2<span class='color_orange'>initial</span>:void</span>; <span class='color_blue'>return = EXN _t$0<span class='color_orange'>update:42</span>:void</span>; <span class='color_blue'>errors = 0<span class='color_orange'>nullify</span>:java.util.List*</span>; <span class='color_blue'>this = @f$1<span class='color_orange'>formal</span>:org.junit.tests.assertion.MultipleFailureExceptionTest*( sub )</span>; <span class='color_blue'>old_this = @f$1<span class='color_orange'>formal</span>:org.junit.tests.assertion.MultipleFailureExceptionTest*( sub )</span>; <span class='color_blue'>e = _t$3<span class='color_orange'>initial</span>:org.junit.tests.assertion.MultipleFailureExceptionTest$ExpectedException*</span> ;
    <span class='color_blue'>exception|->{}<span class='color_orange'>alloc</span>:org.junit.tests.assertion.MultipleFailureExceptionTest$ExpectedException</span>; <span class='color_blue'>_t$0|->{}<span class='color_orange'>alloc</span>:java.lang.AssertionError</span>; <span class='color_blue'>@f$0|->{mIsEmpty:1<span class='color_orange'>update:(z)41</span>}<span class='color_orange'>formal(z)</span>:java.util.List( sub )</span>
    [footprint
       <span class='color_blue'>(@f$2  &lt;=  -1)</span> *
       <span class='color_blue'>$RET_java.util.Collections.singletonList(org.junit.tests.assertion.MultipleFailureExceptionTest$ExpectedException):java.util.List|abducedRetvar = @f$0<span class='color_orange'>none</span>:java.util.List*( sub )</span>; <span class='color_blue'>this = @f$1<span class='color_orange'>formal</span>:org.junit.tests.assertion.MultipleFailureExceptionTest*( sub )</span> ;
       <span class='color_blue'>@f$0|->{mIsEmpty:@f$2<span class='color_orange'>rearrange:(z)27</span>}<span class='color_orange'>rearrange:(z)88</span>:java.util.List( sub )</span>  ]</span>
    
    .... After Symbolic Execution ....
    PROP 1 of 1:
    <span class='color_blue'><span class='color_blue'>(@f$2  &lt;=  -1)</span>; <span class='color_red'>MEMne &lt; new:38 &gt; <span class='color_orange'>{vpath: }</span>(exception)</span>; <span class='color_blue'>MEMne &lt; void Assert.fail():42 &gt; <span class='color_orange'>{vpath: }</span>(_t$0)</span>; <span class='color_blue'>UND &lt; List Collections.singletonList(MultipleFailureExceptionTest$ExpectedException) &gt; :39(@f$0)</span> ;
    <span class='color_blue'>exception = <span class='color_red'>_t$1<span class='color_orange'>update:38</span></span>:java.lang.Throwable*</span>; <span class='color_blue'>CatchVar25 = <span class='color_red'>_t$2<span class='color_orange'>initial</span></span>:void</span>; <span class='color_blue'>return = <span class='color_red'>EXN _t$0<span class='color_orange'>update:42</span></span>:void</span>; <span class='color_blue'>errors = 0<span class='color_orange'>nullify</span>:java.util.List*</span>; <span class='color_blue'>this = @f$1<span class='color_orange'>formal</span>:org.junit.tests.assertion.MultipleFailureExceptionTest*( sub )</span>; <span class='color_blue'>old_this = @f$1<span class='color_orange'>formal</span>:org.junit.tests.assertion.MultipleFailureExceptionTest*( sub )</span>; <span class='color_blue'>e = _t$3<span class='color_orange'>initial</span>:org.junit.tests.assertion.MultipleFailureExceptionTest$ExpectedException*</span> ;
    <span class='color_red'>exception|->{}<span class='color_orange'>alloc</span>:org.junit.tests.assertion.MultipleFailureExceptionTest$ExpectedException</span>; <span class='color_blue'>_t$0|->{}<span class='color_orange'>alloc</span>:java.lang.AssertionError</span>; <span class='color_blue'>@f$0|->{mIsEmpty:1<span class='color_orange'>update:(z)41</span>}<span class='color_orange'>formal(z)</span>:java.util.List( sub )</span>
    [footprint
       <span class='color_blue'>(@f$2  &lt;=  -1)</span> *
       <span class='color_blue'>$RET_java.util.Collections.singletonList(org.junit.tests.assertion.MultipleFailureExceptionTest$ExpectedException):java.util.List|abducedRetvar = @f$0<span class='color_orange'>none</span>:java.util.List*( sub )</span>; <span class='color_blue'>this = @f$1<span class='color_orange'>formal</span>:org.junit.tests.assertion.MultipleFailureExceptionTest*( sub )</span> ;
       <span class='color_blue'>@f$0|->{mIsEmpty:@f$2<span class='color_orange'>rearrange:(z)27</span>}<span class='color_orange'>rearrange:(z)88</span>:java.util.List( sub )</span>  ]</span>
    
    
    
  Processing prop 2/2
    Instruction <span class='color_green'>n$6=_fun_void Assert.fail()() [line 42]</span>
    in vpath_find exp:n$200
    vpath_find: found exception :java.lang.Throwable*
    Found 1 specs for function org.junit.Assert.fail():void
    START EXECUTING SPECS FOR org.junit.Assert.fail():void from state
    <span class='color_blue'>@f$2 = 1 ;
    MEMne &lt; new:38 &gt; <span class='color_orange'>{vpath: }</span>(exception); UND &lt; List Collections.singletonList(MultipleFailureExceptionTest$ExpectedException) &gt; :39(@f$0) ;
    exception = n$200<span class='color_orange'>update:38</span>:java.lang.Throwable*; CatchVar25 = n$201<span class='color_orange'>initial</span>:void; return = n$202<span class='color_orange'>initial</span>:void; errors = 0<span class='color_orange'>nullify</span>:java.util.List*; this = @f$1<span class='color_orange'>formal</span>:org.junit.tests.assertion.MultipleFailureExceptionTest*( sub ); old_this = @f$1<span class='color_orange'>formal</span>:org.junit.tests.assertion.MultipleFailureExceptionTest*( sub ); e = n$203<span class='color_orange'>initial</span>:org.junit.tests.assertion.MultipleFailureExceptionTest$ExpectedException* ;
    exception|->{}<span class='color_orange'>alloc</span>:org.junit.tests.assertion.MultipleFailureExceptionTest$ExpectedException; @f$0|->{mIsEmpty:1<span class='color_orange'>formal(z)</span>}<span class='color_orange'>formal(z)</span>:java.util.List( sub )
    [footprint
       @f$2 = 1 *
       $RET_java.util.Collections.singletonList(org.junit.tests.assertion.MultipleFailureExceptionTest$ExpectedException):java.util.List|abducedRetvar = @f$0<span class='color_orange'>none</span>:java.util.List*( sub ); this = @f$1<span class='color_orange'>formal</span>:org.junit.tests.assertion.MultipleFailureExceptionTest*( sub ) ;
       @f$0|->{mIsEmpty:1<span class='color_orange'>rearrange:(z)27</span>}<span class='color_orange'>rearrange:(z)88</span>:java.util.List( sub )  ]</span>
    EXECUTING SPEC 1/1
    ACTUAL PRECONDITION =
      <span class='color_blue'>@f$2 = 1 ;
      MEMne &lt; new:38 &gt; <span class='color_orange'>{vpath: }</span>(exception); UND &lt; List Collections.singletonList(MultipleFailureExceptionTest$ExpectedException) &gt; :39(@f$0) ;
      exception = n$200<span class='color_orange'>update:38</span>:java.lang.Throwable*; CatchVar25 = n$201<span class='color_orange'>initial</span>:void; return = n$202<span class='color_orange'>initial</span>:void; errors = 0<span class='color_orange'>nullify</span>:java.util.List*; this = @f$1<span class='color_orange'>formal</span>:org.junit.tests.assertion.MultipleFailureExceptionTest*( sub ); old_this = @f$1<span class='color_orange'>formal</span>:org.junit.tests.assertion.MultipleFailureExceptionTest*( sub ); e = n$203<span class='color_orange'>initial</span>:org.junit.tests.assertion.MultipleFailureExceptionTest$ExpectedException* ;
      exception|->{}<span class='color_orange'>alloc</span>:org.junit.tests.assertion.MultipleFailureExceptionTest$ExpectedException; @f$0|->{mIsEmpty:1<span class='color_orange'>formal(z)</span>}<span class='color_orange'>formal(z)</span>:java.util.List( sub )
      [footprint
         @f$2 = 1 *
         $RET_java.util.Collections.singletonList(org.junit.tests.assertion.MultipleFailureExceptionTest$ExpectedException):java.util.List|abducedRetvar = @f$0<span class='color_orange'>none</span>:java.util.List*( sub ); this = @f$1<span class='color_orange'>formal</span>:org.junit.tests.assertion.MultipleFailureExceptionTest*( sub ) ;
         @f$0|->{mIsEmpty:1<span class='color_orange'>rearrange:(z)27</span>}<span class='color_orange'>rearrange:(z)88</span>:java.util.List( sub )  ]</span>
    SPEC =
      --------------------------- ---------- ---------------------------
      PRE:
      <span class='color_blue'><span class='color_blue'></span></span>
      POST 1 of 1:
      <span class='color_blue'><span class='color_red'>MEMne &lt; void Assert.fail(String):96 &gt; <span class='color_orange'>{vpath: }</span>(_t$298)</span> ;
      <span class='color_red'>return|callee = EXN _t$298<span class='color_orange'>update:96</span>:void</span> ;
      <span class='color_red'>_t$298|->{}<span class='color_orange'>alloc</span>:java.lang.AssertionError</span></span>
      ----------------------------------------------------------------
    pre_check_pure_implication
    pi1:
      @f$2 = 1; MEMne &lt; new:38 &gt; <span class='color_orange'>{vpath: }</span>(n$200); UND &lt; List Collections.singletonList(MultipleFailureExceptionTest$ExpectedException) &gt; :39(@f$0)
    pi2:
      
    returns
    sub1:
      @f$2 = 1
    sub2:
      
    Final Implication
    SUB:
      @f$2 = 1
    <span class='color_blue'>MEMne &lt; new:38 &gt; <span class='color_orange'>{vpath: }</span>(exception); UND &lt; List Collections.singletonList(MultipleFailureExceptionTest$ExpectedException) &gt; :39(@f$0); 1 = 1 ;
    exception = n$200<span class='color_orange'>update:38</span>:java.lang.Throwable*; CatchVar25 = n$201<span class='color_orange'>initial</span>:void; return = n$202<span class='color_orange'>initial</span>:void; errors = 0<span class='color_orange'>nullify</span>:java.util.List*; this = @f$1<span class='color_orange'>formal</span>:org.junit.tests.assertion.MultipleFailureExceptionTest*( sub ); old_this = @f$1<span class='color_orange'>formal</span>:org.junit.tests.assertion.MultipleFailureExceptionTest*( sub ); e = n$203<span class='color_orange'>initial</span>:org.junit.tests.assertion.MultipleFailureExceptionTest$ExpectedException* ;
    exception|->{}<span class='color_orange'>alloc</span>:org.junit.tests.assertion.MultipleFailureExceptionTest$ExpectedException; @f$0|->{mIsEmpty:1<span class='color_orange'>formal(z)</span>}<span class='color_orange'>formal(z)</span>:java.util.List( sub )
    [footprint
       1 = 1 *
       $RET_java.util.Collections.singletonList(org.junit.tests.assertion.MultipleFailureExceptionTest$ExpectedException):java.util.List|abducedRetvar = @f$0<span class='color_orange'>none</span>:java.util.List*( sub ); this = @f$1<span class='color_orange'>formal</span>:org.junit.tests.assertion.MultipleFailureExceptionTest*( sub ) ;
       @f$0|->{mIsEmpty:1<span class='color_orange'>rearrange:(z)27</span>}<span class='color_orange'>rearrange:(z)88</span>:java.util.List( sub )  ]</span>
    |-
    <span class='color_blue'></span>
    Result of Abduction
      SUB:
        @f$2 = 1
      <span class='color_blue'>MEMne &lt; new:38 &gt; <span class='color_orange'>{vpath: }</span>(exception); UND &lt; List Collections.singletonList(MultipleFailureExceptionTest$ExpectedException) &gt; :39(@f$0); 1 = 1 ;
      exception = n$200<span class='color_orange'>update:38</span>:java.lang.Throwable*; CatchVar25 = n$201<span class='color_orange'>initial</span>:void; return = n$202<span class='color_orange'>initial</span>:void; errors = 0<span class='color_orange'>nullify</span>:java.util.List*; this = @f$1<span class='color_orange'>formal</span>:org.junit.tests.assertion.MultipleFailureExceptionTest*( sub ); old_this = @f$1<span class='color_orange'>formal</span>:org.junit.tests.assertion.MultipleFailureExceptionTest*( sub ); e = n$203<span class='color_orange'>initial</span>:org.junit.tests.assertion.MultipleFailureExceptionTest$ExpectedException* ;
      exception|->{}<span class='color_orange'>alloc</span>:org.junit.tests.assertion.MultipleFailureExceptionTest$ExpectedException; @f$0|->{mIsEmpty:1<span class='color_orange'>formal(z)</span>}<span class='color_orange'>formal(z)</span>:java.util.List( sub )
      [footprint
         1 = 1 *
         $RET_java.util.Collections.singletonList(org.junit.tests.assertion.MultipleFailureExceptionTest$ExpectedException):java.util.List|abducedRetvar = @f$0<span class='color_orange'>none</span>:java.util.List*( sub ); this = @f$1<span class='color_orange'>formal</span>:org.junit.tests.assertion.MultipleFailureExceptionTest*( sub ) ;
         @f$0|->{mIsEmpty:1<span class='color_orange'>rearrange:(z)27</span>}<span class='color_orange'>rearrange:(z)88</span>:java.util.List( sub )  ]</span>
      |-
      <span class='color_blue'></span>
    returning TRUE
    in vpath_find exp:_t$298
    vpath_find: cannot find _t$298
      New footprint:
      
      Frame fld:
      
      Missing fld:
      
      Instantiated frame:
      n$200|->{}<span class='color_orange'>alloc</span>:org.junit.tests.assertion.MultipleFailureExceptionTest$ExpectedException; @f$0|->{mIsEmpty:1<span class='color_orange'>formal(z)</span>}<span class='color_orange'>formal(z)</span>:java.util.List( sub ); &amp;exception|->n$200<span class='color_orange'>update:38</span>:java.lang.Throwable*; &amp;CatchVar25|->n$201<span class='color_orange'>initial</span>:void; &amp;return|->n$202<span class='color_orange'>initial</span>:void; &amp;errors|->0<span class='color_orange'>nullify</span>:java.util.List*; &amp;this|->@f$1<span class='color_orange'>formal</span>:org.junit.tests.assertion.MultipleFailureExceptionTest*( sub ); &amp;old_this|->@f$1<span class='color_orange'>formal</span>:org.junit.tests.assertion.MultipleFailureExceptionTest*( sub ); &amp;e|->n$203<span class='color_orange'>initial</span>:org.junit.tests.assertion.MultipleFailureExceptionTest$ExpectedException*
      Instantiated post:
      PROP 1 of 1:
      <span class='color_blue'><span class='color_red'>MEMne &lt; void Assert.fail():42 &gt; <span class='color_orange'>{vpath: }</span>(_t$298)</span> ;
      <span class='color_red'>return|callee = EXN _t$298<span class='color_orange'>update:42</span>:void</span> ;
      <span class='color_red'>_t$298|->{}<span class='color_orange'>alloc</span>:java.lang.AssertionError</span></span>
      
    ***** RESULTS FUNCTION CALL *******
    PROP 1 of 1:
    <span class='color_blue'><span class='color_blue'>@f$2 = 1</span> ;
    <span class='color_red'>MEMne &lt; void Assert.fail():42 &gt; <span class='color_orange'>{vpath: }</span>(_t$0)</span>; <span class='color_blue'>MEMne &lt; new:38 &gt; <span class='color_orange'>{vpath: }</span>(exception)</span>; <span class='color_blue'>UND &lt; List Collections.singletonList(MultipleFailureExceptionTest$ExpectedException) &gt; :39(@f$0)</span> ;
    <span class='color_blue'>exception = n$200<span class='color_orange'>update:38</span>:java.lang.Throwable*</span>; <span class='color_blue'>CatchVar25 = n$201<span class='color_orange'>initial</span>:void</span>; <span class='color_blue'>return = <span class='color_red'>EXN _t$0<span class='color_orange'>update:42</span></span>:void</span>; <span class='color_blue'>errors = 0<span class='color_orange'>nullify</span>:java.util.List*</span>; <span class='color_blue'>this = @f$1<span class='color_orange'>formal</span>:org.junit.tests.assertion.MultipleFailureExceptionTest*( sub )</span>; <span class='color_blue'>old_this = @f$1<span class='color_orange'>formal</span>:org.junit.tests.assertion.MultipleFailureExceptionTest*( sub )</span>; <span class='color_blue'>e = n$203<span class='color_orange'>initial</span>:org.junit.tests.assertion.MultipleFailureExceptionTest$ExpectedException*</span> ;
    <span class='color_red'>_t$0|->{}<span class='color_orange'>alloc</span>:java.lang.AssertionError</span>; <span class='color_blue'>exception|->{}<span class='color_orange'>alloc</span>:org.junit.tests.assertion.MultipleFailureExceptionTest$ExpectedException</span>; <span class='color_blue'>@f$0|->{mIsEmpty:1<span class='color_orange'>formal(z)</span>}<span class='color_orange'>formal(z)</span>:java.util.List( sub )</span>
    [footprint
       <span class='color_blue'>@f$2 = 1</span> *
       <span class='color_blue'>$RET_java.util.Collections.singletonList(org.junit.tests.assertion.MultipleFailureExceptionTest$ExpectedException):java.util.List|abducedRetvar = @f$0<span class='color_orange'>none</span>:java.util.List*( sub )</span>; <span class='color_blue'>this = @f$1<span class='color_orange'>formal</span>:org.junit.tests.assertion.MultipleFailureExceptionTest*( sub )</span> ;
       <span class='color_blue'>@f$0|->{mIsEmpty:1<span class='color_orange'>rearrange:(z)27</span>}<span class='color_orange'>rearrange:(z)88</span>:java.util.List( sub )</span>  ]</span>
    ***** END RESULTS FUNCTION CALL *******
    explain_dereference_as_caller_expression @f$0: cannot explain None
    <span class='color_red'>found error in dereference</span>
    spec_pre:
    <span class='color_blue'>@f$2 = 1 ;
    MEMne &lt; void Assert.fail():42 &gt; <span class='color_orange'>{vpath: }</span>(_t$298); MEMne &lt; new:38 &gt; <span class='color_orange'>{vpath: }</span>(exception); UND &lt; List Collections.singletonList(MultipleFailureExceptionTest$ExpectedException) &gt; :39(@f$0) ;
    exception = n$200<span class='color_orange'>update:38</span>:java.lang.Throwable*; CatchVar25 = n$201<span class='color_orange'>initial</span>:void; return = EXN _t$298<span class='color_orange'>update:42</span>:void; errors = 0<span class='color_orange'>nullify</span>:java.util.List*; this = @f$1<span class='color_orange'>formal</span>:org.junit.tests.assertion.MultipleFailureExceptionTest*( sub ); old_this = @f$1<span class='color_orange'>formal</span>:org.junit.tests.assertion.MultipleFailureExceptionTest*( sub ); e = n$203<span class='color_orange'>initial</span>:org.junit.tests.assertion.MultipleFailureExceptionTest$ExpectedException* ;
    _t$298|->{}<span class='color_orange'>alloc</span>:java.lang.AssertionError; exception|->{}<span class='color_orange'>alloc</span>:org.junit.tests.assertion.MultipleFailureExceptionTest$ExpectedException; @f$0|->{mIsEmpty:1<span class='color_orange'>formal(z)</span>}<span class='color_orange'>formal(z)</span>:java.util.List( sub )
    [footprint
       @f$2 = 1 *
       $RET_java.util.Collections.singletonList(org.junit.tests.assertion.MultipleFailureExceptionTest$ExpectedException):java.util.List|abducedRetvar = @f$0<span class='color_orange'>none</span>:java.util.List*( sub ); this = @f$1<span class='color_orange'>formal</span>:org.junit.tests.assertion.MultipleFailureExceptionTest*( sub ) ;
       @f$0|->{mIsEmpty:1<span class='color_orange'>rearrange:(z)27</span>}<span class='color_orange'>rearrange:(z)88</span>:java.util.List( sub )  ]</span>
    exp @f$0 desc: 
    Prover.check_inconsistency_base: inconsistency reason [None]
    Prover.check_inconsistency_base: inconsistency reason [None]
    Prover.check_inconsistency_base: inconsistency reason [None]
    Prover.check_inconsistency_base: inconsistency reason [None]
    Instruction Returns
    PROP 1 of 1:
    <span class='color_blue'><span class='color_blue'>@f$2 = 1</span> ;
    <span class='color_red'>MEMne &lt; new:38 &gt; <span class='color_orange'>{vpath: }</span>(exception)</span>; <span class='color_blue'>MEMne &lt; void Assert.fail():42 &gt; <span class='color_orange'>{vpath: }</span>(_t$0)</span>; <span class='color_blue'>UND &lt; List Collections.singletonList(MultipleFailureExceptionTest$ExpectedException) &gt; :39(@f$0)</span>; <span class='color_red'>RET &lt; void Assert.fail() &gt; (n$6)</span> ;
    <span class='color_blue'>exception = <span class='color_red'>_t$1<span class='color_orange'>update:38</span></span>:java.lang.Throwable*</span>; <span class='color_blue'>CatchVar25 = <span class='color_red'>_t$2<span class='color_orange'>initial</span></span>:void</span>; <span class='color_blue'>return = <span class='color_red'>EXN _t$0<span class='color_orange'>update:42</span></span>:void</span>; <span class='color_blue'>errors = 0<span class='color_orange'>nullify</span>:java.util.List*</span>; <span class='color_blue'>this = @f$1<span class='color_orange'>formal</span>:org.junit.tests.assertion.MultipleFailureExceptionTest*( sub )</span>; <span class='color_blue'>old_this = @f$1<span class='color_orange'>formal</span>:org.junit.tests.assertion.MultipleFailureExceptionTest*( sub )</span>; <span class='color_blue'>e = _t$3<span class='color_orange'>initial</span>:org.junit.tests.assertion.MultipleFailureExceptionTest$ExpectedException*</span> ;
    <span class='color_red'>exception|->{}<span class='color_orange'>alloc</span>:org.junit.tests.assertion.MultipleFailureExceptionTest$ExpectedException</span>; <span class='color_blue'>_t$0|->{}<span class='color_orange'>alloc</span>:java.lang.AssertionError</span>; <span class='color_blue'>@f$0|->{mIsEmpty:1<span class='color_orange'>formal(z)</span>}<span class='color_orange'>formal(z)</span>:java.util.List( sub )</span>
    [footprint
       <span class='color_blue'>@f$2 = 1</span> *
       <span class='color_blue'>$RET_java.util.Collections.singletonList(org.junit.tests.assertion.MultipleFailureExceptionTest$ExpectedException):java.util.List|abducedRetvar = @f$0<span class='color_orange'>none</span>:java.util.List*( sub )</span>; <span class='color_blue'>this = @f$1<span class='color_orange'>formal</span>:org.junit.tests.assertion.MultipleFailureExceptionTest*( sub )</span> ;
       <span class='color_blue'>@f$0|->{mIsEmpty:1<span class='color_orange'>rearrange:(z)27</span>}<span class='color_orange'>rearrange:(z)88</span>:java.util.List( sub )</span>  ]</span>
    
    Instruction <span class='color_green'>EXIT_SCOPE(n$6); [line 42]</span>
    in vpath_find exp:_t$1
    vpath_find: found exception :java.lang.Throwable*
    in vpath_find exp:_t$0
    vpath_find: cannot find _t$0
    Instruction Returns
    PROP 1 of 1:
    <span class='color_blue'><span class='color_blue'>@f$2 = 1</span> ;
    <span class='color_blue'>MEMne &lt; new:38 &gt; <span class='color_orange'>{vpath: }</span>(exception)</span>; <span class='color_blue'>MEMne &lt; void Assert.fail():42 &gt; <span class='color_orange'>{vpath: }</span>(_t$0)</span>; <span class='color_blue'>UND &lt; List Collections.singletonList(MultipleFailureExceptionTest$ExpectedException) &gt; :39(@f$0)</span>; <span class='color_red'>RET &lt; void Assert.fail() &gt; (_t$4)</span> ;
    <span class='color_blue'>exception = _t$1<span class='color_orange'>update:38</span>:java.lang.Throwable*</span>; <span class='color_blue'>CatchVar25 = _t$2<span class='color_orange'>initial</span>:void</span>; <span class='color_blue'>return = EXN _t$0<span class='color_orange'>update:42</span>:void</span>; <span class='color_blue'>errors = 0<span class='color_orange'>nullify</span>:java.util.List*</span>; <span class='color_blue'>this = @f$1<span class='color_orange'>formal</span>:org.junit.tests.assertion.MultipleFailureExceptionTest*( sub )</span>; <span class='color_blue'>old_this = @f$1<span class='color_orange'>formal</span>:org.junit.tests.assertion.MultipleFailureExceptionTest*( sub )</span>; <span class='color_blue'>e = _t$3<span class='color_orange'>initial</span>:org.junit.tests.assertion.MultipleFailureExceptionTest$ExpectedException*</span> ;
    <span class='color_blue'>exception|->{}<span class='color_orange'>alloc</span>:org.junit.tests.assertion.MultipleFailureExceptionTest$ExpectedException</span>; <span class='color_blue'>_t$0|->{}<span class='color_orange'>alloc</span>:java.lang.AssertionError</span>; <span class='color_blue'>@f$0|->{mIsEmpty:1<span class='color_orange'>formal(z)</span>}<span class='color_orange'>formal(z)</span>:java.util.List( sub )</span>
    [footprint
       <span class='color_blue'>@f$2 = 1</span> *
       <span class='color_blue'>$RET_java.util.Collections.singletonList(org.junit.tests.assertion.MultipleFailureExceptionTest$ExpectedException):java.util.List|abducedRetvar = @f$0<span class='color_orange'>none</span>:java.util.List*( sub )</span>; <span class='color_blue'>this = @f$1<span class='color_orange'>formal</span>:org.junit.tests.assertion.MultipleFailureExceptionTest*( sub )</span> ;
       <span class='color_blue'>@f$0|->{mIsEmpty:1<span class='color_orange'>rearrange:(z)27</span>}<span class='color_orange'>rearrange:(z)88</span>:java.util.List( sub )</span>  ]</span>
    
    Instruction <span class='color_green'>APPLY_ABSTRACTION; [line 42]</span>
    in vpath_find exp:_t$1
    vpath_find: found exception :java.lang.Throwable*
    in vpath_find exp:_t$0
    vpath_find: cannot find _t$0
    Prover.check_inconsistency_base: inconsistency reason [None]
    Prover.check_inconsistency_base: inconsistency reason [None]
    Can't find field mIsEmpty in StrexpMatch.find
    Can't find field mIsEmpty in StrexpMatch.find
    <span class='color_blue'>MEMne &lt; new:38 &gt; <span class='color_orange'>{vpath: }</span>(exception); MEMne &lt; void Assert.fail():42 &gt; <span class='color_orange'>{vpath: }</span>(_t$0); UND &lt; List Collections.singletonList(MultipleFailureExceptionTest$ExpectedException) &gt; :39(@f$0) ;
    exception = _t$1<span class='color_orange'>update:38</span>:java.lang.Throwable*; CatchVar25 = _t$2<span class='color_orange'>initial</span>:void; return = EXN _t$0<span class='color_orange'>update:42</span>:void; errors = 0<span class='color_orange'>nullify</span>:java.util.List*; this = @f$1<span class='color_orange'>formal</span>:org.junit.tests.assertion.MultipleFailureExceptionTest*( sub ); old_this = @f$1<span class='color_orange'>formal</span>:org.junit.tests.assertion.MultipleFailureExceptionTest*( sub ); e = _t$3<span class='color_orange'>initial</span>:org.junit.tests.assertion.MultipleFailureExceptionTest$ExpectedException* ;
    exception|->{}<span class='color_orange'>alloc</span>:org.junit.tests.assertion.MultipleFailureExceptionTest$ExpectedException; _t$0|->{}<span class='color_orange'>alloc</span>:java.lang.AssertionError; @f$0|->{mIsEmpty:1<span class='color_orange'>formal(z)</span>}<span class='color_orange'>formal(z)</span>:java.util.List( sub )
    [footprint
       $RET_java.util.Collections.singletonList(org.junit.tests.assertion.MultipleFailureExceptionTest$ExpectedException):java.util.List|abducedRetvar = @f$0<span class='color_orange'>none</span>:java.util.List*( sub ); this = @f$1<span class='color_orange'>formal</span>:org.junit.tests.assertion.MultipleFailureExceptionTest*( sub ) ;
       @f$0|->{mIsEmpty:1<span class='color_orange'>rearrange:(z)27</span>}<span class='color_orange'>rearrange:(z)88</span>:java.util.List( sub )  ]</span>
    Instruction Returns
    PROP 1 of 1:
    <span class='color_blue'><span class='color_blue'>MEMne &lt; new:38 &gt; <span class='color_orange'>{vpath: }</span>(exception)</span>; <span class='color_blue'>MEMne &lt; void Assert.fail():42 &gt; <span class='color_orange'>{vpath: }</span>(_t$0)</span>; <span class='color_blue'>UND &lt; List Collections.singletonList(MultipleFailureExceptionTest$ExpectedException) &gt; :39(@f$0)</span> ;
    <span class='color_blue'>exception = _t$1<span class='color_orange'>update:38</span>:java.lang.Throwable*</span>; <span class='color_blue'>CatchVar25 = _t$2<span class='color_orange'>initial</span>:void</span>; <span class='color_blue'>return = EXN _t$0<span class='color_orange'>update:42</span>:void</span>; <span class='color_blue'>errors = 0<span class='color_orange'>nullify</span>:java.util.List*</span>; <span class='color_blue'>this = @f$1<span class='color_orange'>formal</span>:org.junit.tests.assertion.MultipleFailureExceptionTest*( sub )</span>; <span class='color_blue'>old_this = @f$1<span class='color_orange'>formal</span>:org.junit.tests.assertion.MultipleFailureExceptionTest*( sub )</span>; <span class='color_blue'>e = _t$3<span class='color_orange'>initial</span>:org.junit.tests.assertion.MultipleFailureExceptionTest$ExpectedException*</span> ;
    <span class='color_blue'>exception|->{}<span class='color_orange'>alloc</span>:org.junit.tests.assertion.MultipleFailureExceptionTest$ExpectedException</span>; <span class='color_blue'>_t$0|->{}<span class='color_orange'>alloc</span>:java.lang.AssertionError</span>; <span class='color_blue'>@f$0|->{mIsEmpty:1<span class='color_orange'>formal(z)</span>}<span class='color_orange'>formal(z)</span>:java.util.List( sub )</span>
    [footprint
       <span class='color_blue'>$RET_java.util.Collections.singletonList(org.junit.tests.assertion.MultipleFailureExceptionTest$ExpectedException):java.util.List|abducedRetvar = @f$0<span class='color_orange'>none</span>:java.util.List*( sub )</span>; <span class='color_blue'>this = @f$1<span class='color_orange'>formal</span>:org.junit.tests.assertion.MultipleFailureExceptionTest*( sub )</span> ;
       <span class='color_blue'>@f$0|->{mIsEmpty:1<span class='color_orange'>rearrange:(z)27</span>}<span class='color_orange'>rearrange:(z)88</span>:java.util.List( sub )</span>  ]</span>
    
    .... After Symbolic Execution ....
    PROP 1 of 1:
    <span class='color_blue'><span class='color_red'>MEMne &lt; new:38 &gt; <span class='color_orange'>{vpath: }</span>(exception)</span>; <span class='color_blue'>MEMne &lt; void Assert.fail():42 &gt; <span class='color_orange'>{vpath: }</span>(_t$0)</span>; <span class='color_blue'>UND &lt; List Collections.singletonList(MultipleFailureExceptionTest$ExpectedException) &gt; :39(@f$0)</span> ;
    <span class='color_blue'>exception = <span class='color_red'>_t$1<span class='color_orange'>update:38</span></span>:java.lang.Throwable*</span>; <span class='color_blue'>CatchVar25 = <span class='color_red'>_t$2<span class='color_orange'>initial</span></span>:void</span>; <span class='color_blue'>return = <span class='color_red'>EXN _t$0<span class='color_orange'>update:42</span></span>:void</span>; <span class='color_blue'>errors = 0<span class='color_orange'>nullify</span>:java.util.List*</span>; <span class='color_blue'>this = @f$1<span class='color_orange'>formal</span>:org.junit.tests.assertion.MultipleFailureExceptionTest*( sub )</span>; <span class='color_blue'>old_this = @f$1<span class='color_orange'>formal</span>:org.junit.tests.assertion.MultipleFailureExceptionTest*( sub )</span>; <span class='color_blue'>e = _t$3<span class='color_orange'>initial</span>:org.junit.tests.assertion.MultipleFailureExceptionTest$ExpectedException*</span> ;
    <span class='color_red'>exception|->{}<span class='color_orange'>alloc</span>:org.junit.tests.assertion.MultipleFailureExceptionTest$ExpectedException</span>; <span class='color_blue'>_t$0|->{}<span class='color_orange'>alloc</span>:java.lang.AssertionError</span>; <span class='color_blue'>@f$0|->{mIsEmpty:1<span class='color_orange'>formal(z)</span>}<span class='color_orange'>formal(z)</span>:java.util.List( sub )</span>
    [footprint
       <span class='color_blue'>$RET_java.util.Collections.singletonList(org.junit.tests.assertion.MultipleFailureExceptionTest$ExpectedException):java.util.List|abducedRetvar = @f$0<span class='color_orange'>none</span>:java.util.List*( sub )</span>; <span class='color_blue'>this = @f$1<span class='color_orange'>formal</span>:org.junit.tests.assertion.MultipleFailureExceptionTest*( sub )</span> ;
       <span class='color_blue'>@f$0|->{mIsEmpty:1<span class='color_orange'>rearrange:(z)27</span>}<span class='color_orange'>rearrange:(z)88</span>:java.util.List( sub )</span>  ]</span>
    
    
    
  </LISTING>
<hr width="100%">
<a name="session66" href="./../nodes/org.junit.tests.assertion.MultipleFailureExceptionTest.assertEmptyRethrowsSingleRuntimeException%28%29%3Av.45a2754f89c8ea8a5d2b4deeb88b1440_node7.html#session66">node7#session66</a>(<a href="./../../MultipleFailureExceptionTest.java.5c3e6d8739a7da2006aef2128f5f1965.html#LINE42">42</a>) biabduction
<LISTING class='color_black'>**** [RE:Pending] void MultipleFailureExceptionTest.assertEmptyRethrowsSingleRuntimeException() Node: 7, Procedure: void MultipleFailureExceptionTest.assertEmptyRethrowsSingleRuntimeException(), Todo: 1 ****
  PROP 1 of 1:
  <span class='color_blue'><span class='color_red'>(val$3  &lt;=  -1)</span>; <span class='color_red'>MEMne &lt; new:38 &gt; <span class='color_orange'>{vpath: }</span>(exception)</span>; <span class='color_red'>UND &lt; List Collections.singletonList(MultipleFailureExceptionTest$ExpectedException) &gt; :39($RET_java.util.Collections.singletonList(org.junit.tests.assertion.MultipleFailureExceptionTest$ExpectedException):java.util.List|abducedRetvar)</span> ;
  <span class='color_red'>exception = _t$3<span class='color_orange'>update:38</span>:java.lang.Throwable*</span>; <span class='color_red'>CatchVar25 = _t$2<span class='color_orange'>initial</span>:void</span>; <span class='color_red'>$RET_java.util.Collections.singletonList(org.junit.tests.assertion.MultipleFailureExceptionTest$ExpectedException):java.util.List|abducedRetvar = val$2<span class='color_orange'>formal</span>:java.util.List*( sub )</span>; <span class='color_red'>return = _t$1<span class='color_orange'>initial</span>:void</span>; <span class='color_red'>errors = 0<span class='color_orange'>nullify</span>:java.util.List*</span>; <span class='color_red'>this = val$1<span class='color_orange'>formal</span>:org.junit.tests.assertion.MultipleFailureExceptionTest*( sub )</span>; <span class='color_red'>old_this = val$1<span class='color_orange'>formal</span>:org.junit.tests.assertion.MultipleFailureExceptionTest*( sub )</span>; <span class='color_red'>e = _t$0<span class='color_orange'>initial</span>:org.junit.tests.assertion.MultipleFailureExceptionTest$ExpectedException*</span> ;
  <span class='color_red'>exception|->{}<span class='color_orange'>alloc</span>:org.junit.tests.assertion.MultipleFailureExceptionTest$ExpectedException</span>; <span class='color_red'>$RET_java.util.Collections.singletonList(org.junit.tests.assertion.MultipleFailureExceptionTest$ExpectedException):java.util.List|abducedRetvar|->{mIsEmpty:1<span class='color_orange'>update:(z)41</span>}<span class='color_orange'>formal(z)</span>:java.util.List( sub )</span></span>
  .... Instructions: ....
  <span class='color_green'>n$6=_fun_void Assert.fail()() [line 42];
  EXIT_SCOPE(n$6); [line 42];
  APPLY_ABSTRACTION; [line 42];
  </span>
  
  Processing prop 1/1
    Instruction <span class='color_green'>n$6=_fun_void Assert.fail()() [line 42]</span>
    in vpath_find exp:n$232
    vpath_find: found exception :java.lang.Throwable*
    Found 1 specs for function org.junit.Assert.fail():void
    START EXECUTING SPECS FOR org.junit.Assert.fail():void from state
    <span class='color_blue'>(val$3  &lt;=  -1); MEMne &lt; new:38 &gt; <span class='color_orange'>{vpath: }</span>(exception); UND &lt; List Collections.singletonList(MultipleFailureExceptionTest$ExpectedException) &gt; :39($RET_java.util.Collections.singletonList(org.junit.tests.assertion.MultipleFailureExceptionTest$ExpectedException):java.util.List|abducedRetvar) ;
    exception = n$232<span class='color_orange'>update:38</span>:java.lang.Throwable*; CatchVar25 = n$233<span class='color_orange'>initial</span>:void; $RET_java.util.Collections.singletonList(org.junit.tests.assertion.MultipleFailureExceptionTest$ExpectedException):java.util.List|abducedRetvar = val$2<span class='color_orange'>formal</span>:java.util.List*( sub ); return = n$234<span class='color_orange'>initial</span>:void; errors = 0<span class='color_orange'>nullify</span>:java.util.List*; this = val$1<span class='color_orange'>formal</span>:org.junit.tests.assertion.MultipleFailureExceptionTest*( sub ); old_this = val$1<span class='color_orange'>formal</span>:org.junit.tests.assertion.MultipleFailureExceptionTest*( sub ); e = n$235<span class='color_orange'>initial</span>:org.junit.tests.assertion.MultipleFailureExceptionTest$ExpectedException* ;
    $RET_java.util.Collections.singletonList(org.junit.tests.assertion.MultipleFailureExceptionTest$ExpectedException):java.util.List|abducedRetvar|->{mIsEmpty:1<span class='color_orange'>update:(z)41</span>}<span class='color_orange'>formal(z)</span>:java.util.List( sub ); exception|->{}<span class='color_orange'>alloc</span>:org.junit.tests.assertion.MultipleFailureExceptionTest$ExpectedException</span>
    EXECUTING SPEC 1/1
    ACTUAL PRECONDITION =
      <span class='color_blue'>(val$3  &lt;=  -1); MEMne &lt; new:38 &gt; <span class='color_orange'>{vpath: }</span>(exception); UND &lt; List Collections.singletonList(MultipleFailureExceptionTest$ExpectedException) &gt; :39($RET_java.util.Collections.singletonList(org.junit.tests.assertion.MultipleFailureExceptionTest$ExpectedException):java.util.List|abducedRetvar) ;
      exception = n$232<span class='color_orange'>update:38</span>:java.lang.Throwable*; CatchVar25 = n$233<span class='color_orange'>initial</span>:void; $RET_java.util.Collections.singletonList(org.junit.tests.assertion.MultipleFailureExceptionTest$ExpectedException):java.util.List|abducedRetvar = val$2<span class='color_orange'>formal</span>:java.util.List*( sub ); return = n$234<span class='color_orange'>initial</span>:void; errors = 0<span class='color_orange'>nullify</span>:java.util.List*; this = val$1<span class='color_orange'>formal</span>:org.junit.tests.assertion.MultipleFailureExceptionTest*( sub ); old_this = val$1<span class='color_orange'>formal</span>:org.junit.tests.assertion.MultipleFailureExceptionTest*( sub ); e = n$235<span class='color_orange'>initial</span>:org.junit.tests.assertion.MultipleFailureExceptionTest$ExpectedException* ;
      $RET_java.util.Collections.singletonList(org.junit.tests.assertion.MultipleFailureExceptionTest$ExpectedException):java.util.List|abducedRetvar|->{mIsEmpty:1<span class='color_orange'>update:(z)41</span>}<span class='color_orange'>formal(z)</span>:java.util.List( sub ); exception|->{}<span class='color_orange'>alloc</span>:org.junit.tests.assertion.MultipleFailureExceptionTest$ExpectedException</span>
    SPEC =
      --------------------------- ---------- ---------------------------
      PRE:
      <span class='color_blue'><span class='color_blue'></span></span>
      POST 1 of 1:
      <span class='color_blue'><span class='color_red'>MEMne &lt; void Assert.fail(String):96 &gt; <span class='color_orange'>{vpath: }</span>(_t$365)</span> ;
      <span class='color_red'>return|callee = EXN _t$365<span class='color_orange'>update:96</span>:void</span> ;
      <span class='color_red'>_t$365|->{}<span class='color_orange'>alloc</span>:java.lang.AssertionError</span></span>
      ----------------------------------------------------------------
    pre_check_pure_implication
    pi1:
      (val$3  &lt;=  -1); MEMne &lt; new:38 &gt; <span class='color_orange'>{vpath: }</span>(n$232); UND &lt; List Collections.singletonList(MultipleFailureExceptionTest$ExpectedException) &gt; :39(val$2)
    pi2:
      
    returns
    sub1:
      
    sub2:
      
    Final Implication
    SUB:
      
    <span class='color_blue'>(val$3  &lt;=  -1); MEMne &lt; new:38 &gt; <span class='color_orange'>{vpath: }</span>(exception); UND &lt; List Collections.singletonList(MultipleFailureExceptionTest$ExpectedException) &gt; :39($RET_java.util.Collections.singletonList(org.junit.tests.assertion.MultipleFailureExceptionTest$ExpectedException):java.util.List|abducedRetvar) ;
    exception = n$232<span class='color_orange'>update:38</span>:java.lang.Throwable*; CatchVar25 = n$233<span class='color_orange'>initial</span>:void; $RET_java.util.Collections.singletonList(org.junit.tests.assertion.MultipleFailureExceptionTest$ExpectedException):java.util.List|abducedRetvar = val$2<span class='color_orange'>formal</span>:java.util.List*( sub ); return = n$234<span class='color_orange'>initial</span>:void; errors = 0<span class='color_orange'>nullify</span>:java.util.List*; this = val$1<span class='color_orange'>formal</span>:org.junit.tests.assertion.MultipleFailureExceptionTest*( sub ); old_this = val$1<span class='color_orange'>formal</span>:org.junit.tests.assertion.MultipleFailureExceptionTest*( sub ); e = n$235<span class='color_orange'>initial</span>:org.junit.tests.assertion.MultipleFailureExceptionTest$ExpectedException* ;
    $RET_java.util.Collections.singletonList(org.junit.tests.assertion.MultipleFailureExceptionTest$ExpectedException):java.util.List|abducedRetvar|->{mIsEmpty:1<span class='color_orange'>update:(z)41</span>}<span class='color_orange'>formal(z)</span>:java.util.List( sub ); exception|->{}<span class='color_orange'>alloc</span>:org.junit.tests.assertion.MultipleFailureExceptionTest$ExpectedException</span>
    |-
    <span class='color_blue'></span>
    Result of Abduction
      SUB:
        
      <span class='color_blue'>(val$3  &lt;=  -1); MEMne &lt; new:38 &gt; <span class='color_orange'>{vpath: }</span>(exception); UND &lt; List Collections.singletonList(MultipleFailureExceptionTest$ExpectedException) &gt; :39($RET_java.util.Collections.singletonList(org.junit.tests.assertion.MultipleFailureExceptionTest$ExpectedException):java.util.List|abducedRetvar) ;
      exception = n$232<span class='color_orange'>update:38</span>:java.lang.Throwable*; CatchVar25 = n$233<span class='color_orange'>initial</span>:void; $RET_java.util.Collections.singletonList(org.junit.tests.assertion.MultipleFailureExceptionTest$ExpectedException):java.util.List|abducedRetvar = val$2<span class='color_orange'>formal</span>:java.util.List*( sub ); return = n$234<span class='color_orange'>initial</span>:void; errors = 0<span class='color_orange'>nullify</span>:java.util.List*; this = val$1<span class='color_orange'>formal</span>:org.junit.tests.assertion.MultipleFailureExceptionTest*( sub ); old_this = val$1<span class='color_orange'>formal</span>:org.junit.tests.assertion.MultipleFailureExceptionTest*( sub ); e = n$235<span class='color_orange'>initial</span>:org.junit.tests.assertion.MultipleFailureExceptionTest$ExpectedException* ;
      $RET_java.util.Collections.singletonList(org.junit.tests.assertion.MultipleFailureExceptionTest$ExpectedException):java.util.List|abducedRetvar|->{mIsEmpty:1<span class='color_orange'>update:(z)41</span>}<span class='color_orange'>formal(z)</span>:java.util.List( sub ); exception|->{}<span class='color_orange'>alloc</span>:org.junit.tests.assertion.MultipleFailureExceptionTest$ExpectedException</span>
      |-
      <span class='color_blue'></span>
    returning TRUE
    in vpath_find exp:_t$365
    vpath_find: cannot find _t$365
      New footprint:
      
      Frame fld:
      
      Missing fld:
      
      Instantiated frame:
      val$2|->{mIsEmpty:1<span class='color_orange'>update:(z)41</span>}<span class='color_orange'>formal(z)</span>:java.util.List( sub ); n$232|->{}<span class='color_orange'>alloc</span>:org.junit.tests.assertion.MultipleFailureExceptionTest$ExpectedException; &amp;exception|->n$232<span class='color_orange'>update:38</span>:java.lang.Throwable*; &amp;CatchVar25|->n$233<span class='color_orange'>initial</span>:void; &amp;$RET_java.util.Collections.singletonList(org.junit.tests.assertion.MultipleFailureExceptionTest$ExpectedException):java.util.List|abducedRetvar|->val$2<span class='color_orange'>formal</span>:java.util.List*( sub ); &amp;return|->n$234<span class='color_orange'>initial</span>:void; &amp;errors|->0<span class='color_orange'>nullify</span>:java.util.List*; &amp;this|->val$1<span class='color_orange'>formal</span>:org.junit.tests.assertion.MultipleFailureExceptionTest*( sub ); &amp;old_this|->val$1<span class='color_orange'>formal</span>:org.junit.tests.assertion.MultipleFailureExceptionTest*( sub ); &amp;e|->n$235<span class='color_orange'>initial</span>:org.junit.tests.assertion.MultipleFailureExceptionTest$ExpectedException*
      Instantiated post:
      PROP 1 of 1:
      <span class='color_blue'><span class='color_red'>MEMne &lt; void Assert.fail():42 &gt; <span class='color_orange'>{vpath: }</span>(_t$365)</span> ;
      <span class='color_red'>return|callee = EXN _t$365<span class='color_orange'>update:42</span>:void</span> ;
      <span class='color_red'>_t$365|->{}<span class='color_orange'>alloc</span>:java.lang.AssertionError</span></span>
      
    ***** RESULTS FUNCTION CALL *******
    PROP 1 of 1:
    <span class='color_blue'><span class='color_blue'>(val$3  &lt;=  -1)</span>; <span class='color_red'>MEMne &lt; void Assert.fail():42 &gt; <span class='color_orange'>{vpath: }</span>(_t$0)</span>; <span class='color_blue'>MEMne &lt; new:38 &gt; <span class='color_orange'>{vpath: }</span>(exception)</span>; <span class='color_blue'>UND &lt; List Collections.singletonList(MultipleFailureExceptionTest$ExpectedException) &gt; :39($RET_java.util.Collections.singletonList(org.junit.tests.assertion.MultipleFailureExceptionTest$ExpectedException):java.util.List|abducedRetvar)</span> ;
    <span class='color_blue'>exception = n$232<span class='color_orange'>update:38</span>:java.lang.Throwable*</span>; <span class='color_blue'>CatchVar25 = n$233<span class='color_orange'>initial</span>:void</span>; <span class='color_blue'>$RET_java.util.Collections.singletonList(org.junit.tests.assertion.MultipleFailureExceptionTest$ExpectedException):java.util.List|abducedRetvar = val$2<span class='color_orange'>formal</span>:java.util.List*( sub )</span>; <span class='color_blue'>return = <span class='color_red'>EXN _t$0<span class='color_orange'>update:42</span></span>:void</span>; <span class='color_blue'>errors = 0<span class='color_orange'>nullify</span>:java.util.List*</span>; <span class='color_blue'>this = val$1<span class='color_orange'>formal</span>:org.junit.tests.assertion.MultipleFailureExceptionTest*( sub )</span>; <span class='color_blue'>old_this = val$1<span class='color_orange'>formal</span>:org.junit.tests.assertion.MultipleFailureExceptionTest*( sub )</span>; <span class='color_blue'>e = n$235<span class='color_orange'>initial</span>:org.junit.tests.assertion.MultipleFailureExceptionTest$ExpectedException*</span> ;
    <span class='color_red'>_t$0|->{}<span class='color_orange'>alloc</span>:java.lang.AssertionError</span>; <span class='color_blue'>$RET_java.util.Collections.singletonList(org.junit.tests.assertion.MultipleFailureExceptionTest$ExpectedException):java.util.List|abducedRetvar|->{mIsEmpty:1<span class='color_orange'>update:(z)41</span>}<span class='color_orange'>formal(z)</span>:java.util.List( sub )</span>; <span class='color_blue'>exception|->{}<span class='color_orange'>alloc</span>:org.junit.tests.assertion.MultipleFailureExceptionTest$ExpectedException</span></span>
    ***** END RESULTS FUNCTION CALL *******
    explain_dereference_as_caller_expression val$2: cannot explain None
    <span class='color_red'>found error in dereference</span>
    spec_pre:
    <span class='color_blue'>(val$3  &lt;=  -1); MEMne &lt; void Assert.fail():42 &gt; <span class='color_orange'>{vpath: }</span>(_t$365); MEMne &lt; new:38 &gt; <span class='color_orange'>{vpath: }</span>(exception); UND &lt; List Collections.singletonList(MultipleFailureExceptionTest$ExpectedException) &gt; :39($RET_java.util.Collections.singletonList(org.junit.tests.assertion.MultipleFailureExceptionTest$ExpectedException):java.util.List|abducedRetvar) ;
    exception = n$232<span class='color_orange'>update:38</span>:java.lang.Throwable*; CatchVar25 = n$233<span class='color_orange'>initial</span>:void; $RET_java.util.Collections.singletonList(org.junit.tests.assertion.MultipleFailureExceptionTest$ExpectedException):java.util.List|abducedRetvar = val$2<span class='color_orange'>formal</span>:java.util.List*( sub ); return = EXN _t$365<span class='color_orange'>update:42</span>:void; errors = 0<span class='color_orange'>nullify</span>:java.util.List*; this = val$1<span class='color_orange'>formal</span>:org.junit.tests.assertion.MultipleFailureExceptionTest*( sub ); old_this = val$1<span class='color_orange'>formal</span>:org.junit.tests.assertion.MultipleFailureExceptionTest*( sub ); e = n$235<span class='color_orange'>initial</span>:org.junit.tests.assertion.MultipleFailureExceptionTest$ExpectedException* ;
    _t$365|->{}<span class='color_orange'>alloc</span>:java.lang.AssertionError; $RET_java.util.Collections.singletonList(org.junit.tests.assertion.MultipleFailureExceptionTest$ExpectedException):java.util.List|abducedRetvar|->{mIsEmpty:1<span class='color_orange'>update:(z)41</span>}<span class='color_orange'>formal(z)</span>:java.util.List( sub ); exception|->{}<span class='color_orange'>alloc</span>:org.junit.tests.assertion.MultipleFailureExceptionTest$ExpectedException</span>
    exp val$2 desc: 
    Prover.check_inconsistency_base: inconsistency reason [None]
    Prover.check_inconsistency_base: inconsistency reason [None]
    Prover.check_inconsistency_base: inconsistency reason [None]
    Prover.check_inconsistency_base: inconsistency reason [None]
    Instruction Returns
    PROP 1 of 1:
    <span class='color_blue'><span class='color_blue'>(val$3  &lt;=  -1)</span>; <span class='color_red'>MEMne &lt; new:38 &gt; <span class='color_orange'>{vpath: }</span>(exception)</span>; <span class='color_red'>MEMne &lt; void Assert.fail():42 &gt; <span class='color_orange'>{vpath: }</span>(_t$0)</span>; <span class='color_blue'>UND &lt; List Collections.singletonList(MultipleFailureExceptionTest$ExpectedException) &gt; :39($RET_java.util.Collections.singletonList(org.junit.tests.assertion.MultipleFailureExceptionTest$ExpectedException):java.util.List|abducedRetvar)</span>; <span class='color_red'>RET &lt; void Assert.fail() &gt; (n$6)</span> ;
    <span class='color_blue'>exception = <span class='color_red'>_t$1<span class='color_orange'>update:38</span></span>:java.lang.Throwable*</span>; <span class='color_blue'>CatchVar25 = _t$2<span class='color_orange'>initial</span>:void</span>; <span class='color_blue'>$RET_java.util.Collections.singletonList(org.junit.tests.assertion.MultipleFailureExceptionTest$ExpectedException):java.util.List|abducedRetvar = val$2<span class='color_orange'>formal</span>:java.util.List*( sub )</span>; <span class='color_blue'>return = <span class='color_red'>EXN _t$0<span class='color_orange'>update:42</span></span>:void</span>; <span class='color_blue'>errors = 0<span class='color_orange'>nullify</span>:java.util.List*</span>; <span class='color_blue'>this = val$1<span class='color_orange'>formal</span>:org.junit.tests.assertion.MultipleFailureExceptionTest*( sub )</span>; <span class='color_blue'>old_this = val$1<span class='color_orange'>formal</span>:org.junit.tests.assertion.MultipleFailureExceptionTest*( sub )</span>; <span class='color_blue'>e = <span class='color_red'>_t$3<span class='color_orange'>initial</span></span>:org.junit.tests.assertion.MultipleFailureExceptionTest$ExpectedException*</span> ;
    <span class='color_red'>exception|->{}<span class='color_orange'>alloc</span>:org.junit.tests.assertion.MultipleFailureExceptionTest$ExpectedException</span>; <span class='color_red'>_t$0|->{}<span class='color_orange'>alloc</span>:java.lang.AssertionError</span>; <span class='color_blue'>$RET_java.util.Collections.singletonList(org.junit.tests.assertion.MultipleFailureExceptionTest$ExpectedException):java.util.List|abducedRetvar|->{mIsEmpty:1<span class='color_orange'>update:(z)41</span>}<span class='color_orange'>formal(z)</span>:java.util.List( sub )</span></span>
    
    Instruction <span class='color_green'>EXIT_SCOPE(n$6); [line 42]</span>
    in vpath_find exp:_t$1
    vpath_find: found exception :java.lang.Throwable*
    in vpath_find exp:_t$0
    vpath_find: cannot find _t$0
    Instruction Returns
    PROP 1 of 1:
    <span class='color_blue'><span class='color_blue'>(val$3  &lt;=  -1)</span>; <span class='color_blue'>MEMne &lt; new:38 &gt; <span class='color_orange'>{vpath: }</span>(exception)</span>; <span class='color_blue'>MEMne &lt; void Assert.fail():42 &gt; <span class='color_orange'>{vpath: }</span>(_t$0)</span>; <span class='color_blue'>UND &lt; List Collections.singletonList(MultipleFailureExceptionTest$ExpectedException) &gt; :39($RET_java.util.Collections.singletonList(org.junit.tests.assertion.MultipleFailureExceptionTest$ExpectedException):java.util.List|abducedRetvar)</span>; <span class='color_red'>RET &lt; void Assert.fail() &gt; (_t$4)</span> ;
    <span class='color_blue'>exception = _t$1<span class='color_orange'>update:38</span>:java.lang.Throwable*</span>; <span class='color_blue'>CatchVar25 = _t$2<span class='color_orange'>initial</span>:void</span>; <span class='color_blue'>$RET_java.util.Collections.singletonList(org.junit.tests.assertion.MultipleFailureExceptionTest$ExpectedException):java.util.List|abducedRetvar = val$2<span class='color_orange'>formal</span>:java.util.List*( sub )</span>; <span class='color_blue'>return = EXN _t$0<span class='color_orange'>update:42</span>:void</span>; <span class='color_blue'>errors = 0<span class='color_orange'>nullify</span>:java.util.List*</span>; <span class='color_blue'>this = val$1<span class='color_orange'>formal</span>:org.junit.tests.assertion.MultipleFailureExceptionTest*( sub )</span>; <span class='color_blue'>old_this = val$1<span class='color_orange'>formal</span>:org.junit.tests.assertion.MultipleFailureExceptionTest*( sub )</span>; <span class='color_blue'>e = _t$3<span class='color_orange'>initial</span>:org.junit.tests.assertion.MultipleFailureExceptionTest$ExpectedException*</span> ;
    <span class='color_blue'>exception|->{}<span class='color_orange'>alloc</span>:org.junit.tests.assertion.MultipleFailureExceptionTest$ExpectedException</span>; <span class='color_blue'>_t$0|->{}<span class='color_orange'>alloc</span>:java.lang.AssertionError</span>; <span class='color_blue'>$RET_java.util.Collections.singletonList(org.junit.tests.assertion.MultipleFailureExceptionTest$ExpectedException):java.util.List|abducedRetvar|->{mIsEmpty:1<span class='color_orange'>update:(z)41</span>}<span class='color_orange'>formal(z)</span>:java.util.List( sub )</span></span>
    
    Instruction <span class='color_green'>APPLY_ABSTRACTION; [line 42]</span>
    in vpath_find exp:_t$1
    vpath_find: found exception :java.lang.Throwable*
    in vpath_find exp:_t$0
    vpath_find: cannot find _t$0
    Prover.check_inconsistency_base: inconsistency reason [None]
    Prover.check_inconsistency_base: inconsistency reason [None]
    Can't find field mIsEmpty in StrexpMatch.find
    <span class='color_blue'>MEMne &lt; new:38 &gt; <span class='color_orange'>{vpath: }</span>(exception); MEMne &lt; void Assert.fail():42 &gt; <span class='color_orange'>{vpath: }</span>(_t$0); UND &lt; List Collections.singletonList(MultipleFailureExceptionTest$ExpectedException) &gt; :39($RET_java.util.Collections.singletonList(org.junit.tests.assertion.MultipleFailureExceptionTest$ExpectedException):java.util.List|abducedRetvar) ;
    exception = _t$1<span class='color_orange'>update:38</span>:java.lang.Throwable*; CatchVar25 = _t$2<span class='color_orange'>initial</span>:void; $RET_java.util.Collections.singletonList(org.junit.tests.assertion.MultipleFailureExceptionTest$ExpectedException):java.util.List|abducedRetvar = val$2<span class='color_orange'>formal</span>:java.util.List*( sub ); return = EXN _t$0<span class='color_orange'>update:42</span>:void; errors = 0<span class='color_orange'>nullify</span>:java.util.List*; this = val$1<span class='color_orange'>formal</span>:org.junit.tests.assertion.MultipleFailureExceptionTest*( sub ); old_this = val$1<span class='color_orange'>formal</span>:org.junit.tests.assertion.MultipleFailureExceptionTest*( sub ); e = _t$3<span class='color_orange'>initial</span>:org.junit.tests.assertion.MultipleFailureExceptionTest$ExpectedException* ;
    exception|->{}<span class='color_orange'>alloc</span>:org.junit.tests.assertion.MultipleFailureExceptionTest$ExpectedException; _t$0|->{}<span class='color_orange'>alloc</span>:java.lang.AssertionError; $RET_java.util.Collections.singletonList(org.junit.tests.assertion.MultipleFailureExceptionTest$ExpectedException):java.util.List|abducedRetvar|->{mIsEmpty:1<span class='color_orange'>update:(z)41</span>}<span class='color_orange'>formal(z)</span>:java.util.List( sub )</span>
    Instruction Returns
    PROP 1 of 1:
    <span class='color_blue'><span class='color_blue'>MEMne &lt; new:38 &gt; <span class='color_orange'>{vpath: }</span>(exception)</span>; <span class='color_blue'>MEMne &lt; void Assert.fail():42 &gt; <span class='color_orange'>{vpath: }</span>(_t$0)</span>; <span class='color_blue'>UND &lt; List Collections.singletonList(MultipleFailureExceptionTest$ExpectedException) &gt; :39($RET_java.util.Collections.singletonList(org.junit.tests.assertion.MultipleFailureExceptionTest$ExpectedException):java.util.List|abducedRetvar)</span> ;
    <span class='color_blue'>exception = _t$1<span class='color_orange'>update:38</span>:java.lang.Throwable*</span>; <span class='color_blue'>CatchVar25 = _t$2<span class='color_orange'>initial</span>:void</span>; <span class='color_blue'>$RET_java.util.Collections.singletonList(org.junit.tests.assertion.MultipleFailureExceptionTest$ExpectedException):java.util.List|abducedRetvar = val$2<span class='color_orange'>formal</span>:java.util.List*( sub )</span>; <span class='color_blue'>return = EXN _t$0<span class='color_orange'>update:42</span>:void</span>; <span class='color_blue'>errors = 0<span class='color_orange'>nullify</span>:java.util.List*</span>; <span class='color_blue'>this = val$1<span class='color_orange'>formal</span>:org.junit.tests.assertion.MultipleFailureExceptionTest*( sub )</span>; <span class='color_blue'>old_this = val$1<span class='color_orange'>formal</span>:org.junit.tests.assertion.MultipleFailureExceptionTest*( sub )</span>; <span class='color_blue'>e = _t$3<span class='color_orange'>initial</span>:org.junit.tests.assertion.MultipleFailureExceptionTest$ExpectedException*</span> ;
    <span class='color_blue'>exception|->{}<span class='color_orange'>alloc</span>:org.junit.tests.assertion.MultipleFailureExceptionTest$ExpectedException</span>; <span class='color_blue'>_t$0|->{}<span class='color_orange'>alloc</span>:java.lang.AssertionError</span>; <span class='color_blue'>$RET_java.util.Collections.singletonList(org.junit.tests.assertion.MultipleFailureExceptionTest$ExpectedException):java.util.List|abducedRetvar|->{mIsEmpty:1<span class='color_orange'>update:(z)41</span>}<span class='color_orange'>formal(z)</span>:java.util.List( sub )</span></span>
    
    .... After Symbolic Execution ....
    PROP 1 of 1:
    <span class='color_blue'><span class='color_red'>MEMne &lt; new:38 &gt; <span class='color_orange'>{vpath: }</span>(exception)</span>; <span class='color_red'>MEMne &lt; void Assert.fail():42 &gt; <span class='color_orange'>{vpath: }</span>(_t$0)</span>; <span class='color_blue'>UND &lt; List Collections.singletonList(MultipleFailureExceptionTest$ExpectedException) &gt; :39($RET_java.util.Collections.singletonList(org.junit.tests.assertion.MultipleFailureExceptionTest$ExpectedException):java.util.List|abducedRetvar)</span> ;
    <span class='color_blue'>exception = <span class='color_red'>_t$1<span class='color_orange'>update:38</span></span>:java.lang.Throwable*</span>; <span class='color_blue'>CatchVar25 = _t$2<span class='color_orange'>initial</span>:void</span>; <span class='color_blue'>$RET_java.util.Collections.singletonList(org.junit.tests.assertion.MultipleFailureExceptionTest$ExpectedException):java.util.List|abducedRetvar = val$2<span class='color_orange'>formal</span>:java.util.List*( sub )</span>; <span class='color_blue'>return = <span class='color_red'>EXN _t$0<span class='color_orange'>update:42</span></span>:void</span>; <span class='color_blue'>errors = 0<span class='color_orange'>nullify</span>:java.util.List*</span>; <span class='color_blue'>this = val$1<span class='color_orange'>formal</span>:org.junit.tests.assertion.MultipleFailureExceptionTest*( sub )</span>; <span class='color_blue'>old_this = val$1<span class='color_orange'>formal</span>:org.junit.tests.assertion.MultipleFailureExceptionTest*( sub )</span>; <span class='color_blue'>e = <span class='color_red'>_t$3<span class='color_orange'>initial</span></span>:org.junit.tests.assertion.MultipleFailureExceptionTest$ExpectedException*</span> ;
    <span class='color_red'>exception|->{}<span class='color_orange'>alloc</span>:org.junit.tests.assertion.MultipleFailureExceptionTest$ExpectedException</span>; <span class='color_red'>_t$0|->{}<span class='color_orange'>alloc</span>:java.lang.AssertionError</span>; <span class='color_blue'>$RET_java.util.Collections.singletonList(org.junit.tests.assertion.MultipleFailureExceptionTest$ExpectedException):java.util.List|abducedRetvar|->{mIsEmpty:1<span class='color_orange'>update:(z)41</span>}<span class='color_orange'>formal(z)</span>:java.util.List( sub )</span></span>
    
    
    
  </LISTING>
<hr width="100%">
<a name="session78" href="./../nodes/org.junit.tests.assertion.MultipleFailureExceptionTest.assertEmptyRethrowsSingleRuntimeException%28%29%3Av.45a2754f89c8ea8a5d2b4deeb88b1440_node7.html#session78">node7#session78</a>(<a href="./../../MultipleFailureExceptionTest.java.5c3e6d8739a7da2006aef2128f5f1965.html#LINE42">42</a>) biabduction
<LISTING class='color_black'>**** [RE:Pending] void MultipleFailureExceptionTest.assertEmptyRethrowsSingleRuntimeException() Node: 7, Procedure: void MultipleFailureExceptionTest.assertEmptyRethrowsSingleRuntimeException(), Todo: 1 ****
  PROP 1 of 1:
  <span class='color_blue'><span class='color_red'>MEMne &lt; new:38 &gt; <span class='color_orange'>{vpath: }</span>(exception)</span>; <span class='color_red'>UND &lt; List Collections.singletonList(MultipleFailureExceptionTest$ExpectedException) &gt; :39($RET_java.util.Collections.singletonList(org.junit.tests.assertion.MultipleFailureExceptionTest$ExpectedException):java.util.List|abducedRetvar)</span> ;
  <span class='color_red'>exception = _t$3<span class='color_orange'>update:38</span>:java.lang.Throwable*</span>; <span class='color_red'>CatchVar25 = _t$2<span class='color_orange'>initial</span>:void</span>; <span class='color_red'>$RET_java.util.Collections.singletonList(org.junit.tests.assertion.MultipleFailureExceptionTest$ExpectedException):java.util.List|abducedRetvar = val$2<span class='color_orange'>formal</span>:java.util.List*( sub )</span>; <span class='color_red'>return = _t$1<span class='color_orange'>initial</span>:void</span>; <span class='color_red'>errors = 0<span class='color_orange'>nullify</span>:java.util.List*</span>; <span class='color_red'>this = val$1<span class='color_orange'>formal</span>:org.junit.tests.assertion.MultipleFailureExceptionTest*( sub )</span>; <span class='color_red'>old_this = val$1<span class='color_orange'>formal</span>:org.junit.tests.assertion.MultipleFailureExceptionTest*( sub )</span>; <span class='color_red'>e = _t$0<span class='color_orange'>initial</span>:org.junit.tests.assertion.MultipleFailureExceptionTest$ExpectedException*</span> ;
  <span class='color_red'>exception|->{}<span class='color_orange'>alloc</span>:org.junit.tests.assertion.MultipleFailureExceptionTest$ExpectedException</span>; <span class='color_red'>$RET_java.util.Collections.singletonList(org.junit.tests.assertion.MultipleFailureExceptionTest$ExpectedException):java.util.List|abducedRetvar|->{mIsEmpty:1<span class='color_orange'>formal(z)</span>}<span class='color_orange'>formal(z)</span>:java.util.List( sub )</span></span>
  .... Instructions: ....
  <span class='color_green'>n$6=_fun_void Assert.fail()() [line 42];
  EXIT_SCOPE(n$6); [line 42];
  APPLY_ABSTRACTION; [line 42];
  </span>
  
  Processing prop 1/1
    Instruction <span class='color_green'>n$6=_fun_void Assert.fail()() [line 42]</span>
    in vpath_find exp:n$299
    vpath_find: found exception :java.lang.Throwable*
    Found 1 specs for function org.junit.Assert.fail():void
    START EXECUTING SPECS FOR org.junit.Assert.fail():void from state
    <span class='color_blue'>MEMne &lt; new:38 &gt; <span class='color_orange'>{vpath: }</span>(exception); UND &lt; List Collections.singletonList(MultipleFailureExceptionTest$ExpectedException) &gt; :39($RET_java.util.Collections.singletonList(org.junit.tests.assertion.MultipleFailureExceptionTest$ExpectedException):java.util.List|abducedRetvar) ;
    exception = n$299<span class='color_orange'>update:38</span>:java.lang.Throwable*; CatchVar25 = n$300<span class='color_orange'>initial</span>:void; $RET_java.util.Collections.singletonList(org.junit.tests.assertion.MultipleFailureExceptionTest$ExpectedException):java.util.List|abducedRetvar = val$2<span class='color_orange'>formal</span>:java.util.List*( sub ); return = n$301<span class='color_orange'>initial</span>:void; errors = 0<span class='color_orange'>nullify</span>:java.util.List*; this = val$1<span class='color_orange'>formal</span>:org.junit.tests.assertion.MultipleFailureExceptionTest*( sub ); old_this = val$1<span class='color_orange'>formal</span>:org.junit.tests.assertion.MultipleFailureExceptionTest*( sub ); e = n$302<span class='color_orange'>initial</span>:org.junit.tests.assertion.MultipleFailureExceptionTest$ExpectedException* ;
    $RET_java.util.Collections.singletonList(org.junit.tests.assertion.MultipleFailureExceptionTest$ExpectedException):java.util.List|abducedRetvar|->{mIsEmpty:1<span class='color_orange'>formal(z)</span>}<span class='color_orange'>formal(z)</span>:java.util.List( sub ); exception|->{}<span class='color_orange'>alloc</span>:org.junit.tests.assertion.MultipleFailureExceptionTest$ExpectedException</span>
    EXECUTING SPEC 1/1
    ACTUAL PRECONDITION =
      <span class='color_blue'>MEMne &lt; new:38 &gt; <span class='color_orange'>{vpath: }</span>(exception); UND &lt; List Collections.singletonList(MultipleFailureExceptionTest$ExpectedException) &gt; :39($RET_java.util.Collections.singletonList(org.junit.tests.assertion.MultipleFailureExceptionTest$ExpectedException):java.util.List|abducedRetvar) ;
      exception = n$299<span class='color_orange'>update:38</span>:java.lang.Throwable*; CatchVar25 = n$300<span class='color_orange'>initial</span>:void; $RET_java.util.Collections.singletonList(org.junit.tests.assertion.MultipleFailureExceptionTest$ExpectedException):java.util.List|abducedRetvar = val$2<span class='color_orange'>formal</span>:java.util.List*( sub ); return = n$301<span class='color_orange'>initial</span>:void; errors = 0<span class='color_orange'>nullify</span>:java.util.List*; this = val$1<span class='color_orange'>formal</span>:org.junit.tests.assertion.MultipleFailureExceptionTest*( sub ); old_this = val$1<span class='color_orange'>formal</span>:org.junit.tests.assertion.MultipleFailureExceptionTest*( sub ); e = n$302<span class='color_orange'>initial</span>:org.junit.tests.assertion.MultipleFailureExceptionTest$ExpectedException* ;
      $RET_java.util.Collections.singletonList(org.junit.tests.assertion.MultipleFailureExceptionTest$ExpectedException):java.util.List|abducedRetvar|->{mIsEmpty:1<span class='color_orange'>formal(z)</span>}<span class='color_orange'>formal(z)</span>:java.util.List( sub ); exception|->{}<span class='color_orange'>alloc</span>:org.junit.tests.assertion.MultipleFailureExceptionTest$ExpectedException</span>
    SPEC =
      --------------------------- ---------- ---------------------------
      PRE:
      <span class='color_blue'><span class='color_blue'></span></span>
      POST 1 of 1:
      <span class='color_blue'><span class='color_red'>MEMne &lt; void Assert.fail(String):96 &gt; <span class='color_orange'>{vpath: }</span>(_t$491)</span> ;
      <span class='color_red'>return|callee = EXN _t$491<span class='color_orange'>update:96</span>:void</span> ;
      <span class='color_red'>_t$491|->{}<span class='color_orange'>alloc</span>:java.lang.AssertionError</span></span>
      ----------------------------------------------------------------
    pre_check_pure_implication
    pi1:
      MEMne &lt; new:38 &gt; <span class='color_orange'>{vpath: }</span>(n$299); UND &lt; List Collections.singletonList(MultipleFailureExceptionTest$ExpectedException) &gt; :39(val$2)
    pi2:
      
    returns
    sub1:
      
    sub2:
      
    Final Implication
    SUB:
      
    <span class='color_blue'>MEMne &lt; new:38 &gt; <span class='color_orange'>{vpath: }</span>(exception); UND &lt; List Collections.singletonList(MultipleFailureExceptionTest$ExpectedException) &gt; :39($RET_java.util.Collections.singletonList(org.junit.tests.assertion.MultipleFailureExceptionTest$ExpectedException):java.util.List|abducedRetvar) ;
    exception = n$299<span class='color_orange'>update:38</span>:java.lang.Throwable*; CatchVar25 = n$300<span class='color_orange'>initial</span>:void; $RET_java.util.Collections.singletonList(org.junit.tests.assertion.MultipleFailureExceptionTest$ExpectedException):java.util.List|abducedRetvar = val$2<span class='color_orange'>formal</span>:java.util.List*( sub ); return = n$301<span class='color_orange'>initial</span>:void; errors = 0<span class='color_orange'>nullify</span>:java.util.List*; this = val$1<span class='color_orange'>formal</span>:org.junit.tests.assertion.MultipleFailureExceptionTest*( sub ); old_this = val$1<span class='color_orange'>formal</span>:org.junit.tests.assertion.MultipleFailureExceptionTest*( sub ); e = n$302<span class='color_orange'>initial</span>:org.junit.tests.assertion.MultipleFailureExceptionTest$ExpectedException* ;
    $RET_java.util.Collections.singletonList(org.junit.tests.assertion.MultipleFailureExceptionTest$ExpectedException):java.util.List|abducedRetvar|->{mIsEmpty:1<span class='color_orange'>formal(z)</span>}<span class='color_orange'>formal(z)</span>:java.util.List( sub ); exception|->{}<span class='color_orange'>alloc</span>:org.junit.tests.assertion.MultipleFailureExceptionTest$ExpectedException</span>
    |-
    <span class='color_blue'></span>
    Result of Abduction
      SUB:
        
      <span class='color_blue'>MEMne &lt; new:38 &gt; <span class='color_orange'>{vpath: }</span>(exception); UND &lt; List Collections.singletonList(MultipleFailureExceptionTest$ExpectedException) &gt; :39($RET_java.util.Collections.singletonList(org.junit.tests.assertion.MultipleFailureExceptionTest$ExpectedException):java.util.List|abducedRetvar) ;
      exception = n$299<span class='color_orange'>update:38</span>:java.lang.Throwable*; CatchVar25 = n$300<span class='color_orange'>initial</span>:void; $RET_java.util.Collections.singletonList(org.junit.tests.assertion.MultipleFailureExceptionTest$ExpectedException):java.util.List|abducedRetvar = val$2<span class='color_orange'>formal</span>:java.util.List*( sub ); return = n$301<span class='color_orange'>initial</span>:void; errors = 0<span class='color_orange'>nullify</span>:java.util.List*; this = val$1<span class='color_orange'>formal</span>:org.junit.tests.assertion.MultipleFailureExceptionTest*( sub ); old_this = val$1<span class='color_orange'>formal</span>:org.junit.tests.assertion.MultipleFailureExceptionTest*( sub ); e = n$302<span class='color_orange'>initial</span>:org.junit.tests.assertion.MultipleFailureExceptionTest$ExpectedException* ;
      $RET_java.util.Collections.singletonList(org.junit.tests.assertion.MultipleFailureExceptionTest$ExpectedException):java.util.List|abducedRetvar|->{mIsEmpty:1<span class='color_orange'>formal(z)</span>}<span class='color_orange'>formal(z)</span>:java.util.List( sub ); exception|->{}<span class='color_orange'>alloc</span>:org.junit.tests.assertion.MultipleFailureExceptionTest$ExpectedException</span>
      |-
      <span class='color_blue'></span>
    returning TRUE
    in vpath_find exp:_t$491
    vpath_find: cannot find _t$491
      New footprint:
      
      Frame fld:
      
      Missing fld:
      
      Instantiated frame:
      val$2|->{mIsEmpty:1<span class='color_orange'>formal(z)</span>}<span class='color_orange'>formal(z)</span>:java.util.List( sub ); n$299|->{}<span class='color_orange'>alloc</span>:org.junit.tests.assertion.MultipleFailureExceptionTest$ExpectedException; &amp;exception|->n$299<span class='color_orange'>update:38</span>:java.lang.Throwable*; &amp;CatchVar25|->n$300<span class='color_orange'>initial</span>:void; &amp;$RET_java.util.Collections.singletonList(org.junit.tests.assertion.MultipleFailureExceptionTest$ExpectedException):java.util.List|abducedRetvar|->val$2<span class='color_orange'>formal</span>:java.util.List*( sub ); &amp;return|->n$301<span class='color_orange'>initial</span>:void; &amp;errors|->0<span class='color_orange'>nullify</span>:java.util.List*; &amp;this|->val$1<span class='color_orange'>formal</span>:org.junit.tests.assertion.MultipleFailureExceptionTest*( sub ); &amp;old_this|->val$1<span class='color_orange'>formal</span>:org.junit.tests.assertion.MultipleFailureExceptionTest*( sub ); &amp;e|->n$302<span class='color_orange'>initial</span>:org.junit.tests.assertion.MultipleFailureExceptionTest$ExpectedException*
      Instantiated post:
      PROP 1 of 1:
      <span class='color_blue'><span class='color_red'>MEMne &lt; void Assert.fail():42 &gt; <span class='color_orange'>{vpath: }</span>(_t$491)</span> ;
      <span class='color_red'>return|callee = EXN _t$491<span class='color_orange'>update:42</span>:void</span> ;
      <span class='color_red'>_t$491|->{}<span class='color_orange'>alloc</span>:java.lang.AssertionError</span></span>
      
    ***** RESULTS FUNCTION CALL *******
    PROP 1 of 1:
    <span class='color_blue'><span class='color_red'>MEMne &lt; void Assert.fail():42 &gt; <span class='color_orange'>{vpath: }</span>(_t$0)</span>; <span class='color_blue'>MEMne &lt; new:38 &gt; <span class='color_orange'>{vpath: }</span>(exception)</span>; <span class='color_blue'>UND &lt; List Collections.singletonList(MultipleFailureExceptionTest$ExpectedException) &gt; :39($RET_java.util.Collections.singletonList(org.junit.tests.assertion.MultipleFailureExceptionTest$ExpectedException):java.util.List|abducedRetvar)</span> ;
    <span class='color_blue'>exception = n$299<span class='color_orange'>update:38</span>:java.lang.Throwable*</span>; <span class='color_blue'>CatchVar25 = n$300<span class='color_orange'>initial</span>:void</span>; <span class='color_blue'>$RET_java.util.Collections.singletonList(org.junit.tests.assertion.MultipleFailureExceptionTest$ExpectedException):java.util.List|abducedRetvar = val$2<span class='color_orange'>formal</span>:java.util.List*( sub )</span>; <span class='color_blue'>return = <span class='color_red'>EXN _t$0<span class='color_orange'>update:42</span></span>:void</span>; <span class='color_blue'>errors = 0<span class='color_orange'>nullify</span>:java.util.List*</span>; <span class='color_blue'>this = val$1<span class='color_orange'>formal</span>:org.junit.tests.assertion.MultipleFailureExceptionTest*( sub )</span>; <span class='color_blue'>old_this = val$1<span class='color_orange'>formal</span>:org.junit.tests.assertion.MultipleFailureExceptionTest*( sub )</span>; <span class='color_blue'>e = n$302<span class='color_orange'>initial</span>:org.junit.tests.assertion.MultipleFailureExceptionTest$ExpectedException*</span> ;
    <span class='color_red'>_t$0|->{}<span class='color_orange'>alloc</span>:java.lang.AssertionError</span>; <span class='color_blue'>$RET_java.util.Collections.singletonList(org.junit.tests.assertion.MultipleFailureExceptionTest$ExpectedException):java.util.List|abducedRetvar|->{mIsEmpty:1<span class='color_orange'>formal(z)</span>}<span class='color_orange'>formal(z)</span>:java.util.List( sub )</span>; <span class='color_blue'>exception|->{}<span class='color_orange'>alloc</span>:org.junit.tests.assertion.MultipleFailureExceptionTest$ExpectedException</span></span>
    ***** END RESULTS FUNCTION CALL *******
    explain_dereference_as_caller_expression val$2: cannot explain None
    <span class='color_red'>found error in dereference</span>
    spec_pre:
    <span class='color_blue'>MEMne &lt; void Assert.fail():42 &gt; <span class='color_orange'>{vpath: }</span>(_t$491); MEMne &lt; new:38 &gt; <span class='color_orange'>{vpath: }</span>(exception); UND &lt; List Collections.singletonList(MultipleFailureExceptionTest$ExpectedException) &gt; :39($RET_java.util.Collections.singletonList(org.junit.tests.assertion.MultipleFailureExceptionTest$ExpectedException):java.util.List|abducedRetvar) ;
    exception = n$299<span class='color_orange'>update:38</span>:java.lang.Throwable*; CatchVar25 = n$300<span class='color_orange'>initial</span>:void; $RET_java.util.Collections.singletonList(org.junit.tests.assertion.MultipleFailureExceptionTest$ExpectedException):java.util.List|abducedRetvar = val$2<span class='color_orange'>formal</span>:java.util.List*( sub ); return = EXN _t$491<span class='color_orange'>update:42</span>:void; errors = 0<span class='color_orange'>nullify</span>:java.util.List*; this = val$1<span class='color_orange'>formal</span>:org.junit.tests.assertion.MultipleFailureExceptionTest*( sub ); old_this = val$1<span class='color_orange'>formal</span>:org.junit.tests.assertion.MultipleFailureExceptionTest*( sub ); e = n$302<span class='color_orange'>initial</span>:org.junit.tests.assertion.MultipleFailureExceptionTest$ExpectedException* ;
    _t$491|->{}<span class='color_orange'>alloc</span>:java.lang.AssertionError; $RET_java.util.Collections.singletonList(org.junit.tests.assertion.MultipleFailureExceptionTest$ExpectedException):java.util.List|abducedRetvar|->{mIsEmpty:1<span class='color_orange'>formal(z)</span>}<span class='color_orange'>formal(z)</span>:java.util.List( sub ); exception|->{}<span class='color_orange'>alloc</span>:org.junit.tests.assertion.MultipleFailureExceptionTest$ExpectedException</span>
    exp val$2 desc: 
    Prover.check_inconsistency_base: inconsistency reason [None]
    Prover.check_inconsistency_base: inconsistency reason [None]
    Prover.check_inconsistency_base: inconsistency reason [None]
    Prover.check_inconsistency_base: inconsistency reason [None]
    Instruction Returns
    PROP 1 of 1:
    <span class='color_blue'><span class='color_red'>MEMne &lt; new:38 &gt; <span class='color_orange'>{vpath: }</span>(exception)</span>; <span class='color_red'>MEMne &lt; void Assert.fail():42 &gt; <span class='color_orange'>{vpath: }</span>(_t$0)</span>; <span class='color_blue'>UND &lt; List Collections.singletonList(MultipleFailureExceptionTest$ExpectedException) &gt; :39($RET_java.util.Collections.singletonList(org.junit.tests.assertion.MultipleFailureExceptionTest$ExpectedException):java.util.List|abducedRetvar)</span>; <span class='color_red'>RET &lt; void Assert.fail() &gt; (n$6)</span> ;
    <span class='color_blue'>exception = <span class='color_red'>_t$1<span class='color_orange'>update:38</span></span>:java.lang.Throwable*</span>; <span class='color_blue'>CatchVar25 = _t$2<span class='color_orange'>initial</span>:void</span>; <span class='color_blue'>$RET_java.util.Collections.singletonList(org.junit.tests.assertion.MultipleFailureExceptionTest$ExpectedException):java.util.List|abducedRetvar = val$2<span class='color_orange'>formal</span>:java.util.List*( sub )</span>; <span class='color_blue'>return = <span class='color_red'>EXN _t$0<span class='color_orange'>update:42</span></span>:void</span>; <span class='color_blue'>errors = 0<span class='color_orange'>nullify</span>:java.util.List*</span>; <span class='color_blue'>this = val$1<span class='color_orange'>formal</span>:org.junit.tests.assertion.MultipleFailureExceptionTest*( sub )</span>; <span class='color_blue'>old_this = val$1<span class='color_orange'>formal</span>:org.junit.tests.assertion.MultipleFailureExceptionTest*( sub )</span>; <span class='color_blue'>e = <span class='color_red'>_t$3<span class='color_orange'>initial</span></span>:org.junit.tests.assertion.MultipleFailureExceptionTest$ExpectedException*</span> ;
    <span class='color_red'>exception|->{}<span class='color_orange'>alloc</span>:org.junit.tests.assertion.MultipleFailureExceptionTest$ExpectedException</span>; <span class='color_red'>_t$0|->{}<span class='color_orange'>alloc</span>:java.lang.AssertionError</span>; <span class='color_blue'>$RET_java.util.Collections.singletonList(org.junit.tests.assertion.MultipleFailureExceptionTest$ExpectedException):java.util.List|abducedRetvar|->{mIsEmpty:1<span class='color_orange'>formal(z)</span>}<span class='color_orange'>formal(z)</span>:java.util.List( sub )</span></span>
    
    Instruction <span class='color_green'>EXIT_SCOPE(n$6); [line 42]</span>
    in vpath_find exp:_t$1
    vpath_find: found exception :java.lang.Throwable*
    in vpath_find exp:_t$0
    vpath_find: cannot find _t$0
    Instruction Returns
    PROP 1 of 1:
    <span class='color_blue'><span class='color_blue'>MEMne &lt; new:38 &gt; <span class='color_orange'>{vpath: }</span>(exception)</span>; <span class='color_blue'>MEMne &lt; void Assert.fail():42 &gt; <span class='color_orange'>{vpath: }</span>(_t$0)</span>; <span class='color_blue'>UND &lt; List Collections.singletonList(MultipleFailureExceptionTest$ExpectedException) &gt; :39($RET_java.util.Collections.singletonList(org.junit.tests.assertion.MultipleFailureExceptionTest$ExpectedException):java.util.List|abducedRetvar)</span>; <span class='color_red'>RET &lt; void Assert.fail() &gt; (_t$4)</span> ;
    <span class='color_blue'>exception = _t$1<span class='color_orange'>update:38</span>:java.lang.Throwable*</span>; <span class='color_blue'>CatchVar25 = _t$2<span class='color_orange'>initial</span>:void</span>; <span class='color_blue'>$RET_java.util.Collections.singletonList(org.junit.tests.assertion.MultipleFailureExceptionTest$ExpectedException):java.util.List|abducedRetvar = val$2<span class='color_orange'>formal</span>:java.util.List*( sub )</span>; <span class='color_blue'>return = EXN _t$0<span class='color_orange'>update:42</span>:void</span>; <span class='color_blue'>errors = 0<span class='color_orange'>nullify</span>:java.util.List*</span>; <span class='color_blue'>this = val$1<span class='color_orange'>formal</span>:org.junit.tests.assertion.MultipleFailureExceptionTest*( sub )</span>; <span class='color_blue'>old_this = val$1<span class='color_orange'>formal</span>:org.junit.tests.assertion.MultipleFailureExceptionTest*( sub )</span>; <span class='color_blue'>e = _t$3<span class='color_orange'>initial</span>:org.junit.tests.assertion.MultipleFailureExceptionTest$ExpectedException*</span> ;
    <span class='color_blue'>exception|->{}<span class='color_orange'>alloc</span>:org.junit.tests.assertion.MultipleFailureExceptionTest$ExpectedException</span>; <span class='color_blue'>_t$0|->{}<span class='color_orange'>alloc</span>:java.lang.AssertionError</span>; <span class='color_blue'>$RET_java.util.Collections.singletonList(org.junit.tests.assertion.MultipleFailureExceptionTest$ExpectedException):java.util.List|abducedRetvar|->{mIsEmpty:1<span class='color_orange'>formal(z)</span>}<span class='color_orange'>formal(z)</span>:java.util.List( sub )</span></span>
    
    Instruction <span class='color_green'>APPLY_ABSTRACTION; [line 42]</span>
    in vpath_find exp:_t$1
    vpath_find: found exception :java.lang.Throwable*
    in vpath_find exp:_t$0
    vpath_find: cannot find _t$0
    Prover.check_inconsistency_base: inconsistency reason [None]
    Prover.check_inconsistency_base: inconsistency reason [None]
    Can't find field mIsEmpty in StrexpMatch.find
    <span class='color_blue'>MEMne &lt; new:38 &gt; <span class='color_orange'>{vpath: }</span>(exception); MEMne &lt; void Assert.fail():42 &gt; <span class='color_orange'>{vpath: }</span>(_t$0); UND &lt; List Collections.singletonList(MultipleFailureExceptionTest$ExpectedException) &gt; :39($RET_java.util.Collections.singletonList(org.junit.tests.assertion.MultipleFailureExceptionTest$ExpectedException):java.util.List|abducedRetvar) ;
    exception = _t$1<span class='color_orange'>update:38</span>:java.lang.Throwable*; CatchVar25 = _t$2<span class='color_orange'>initial</span>:void; $RET_java.util.Collections.singletonList(org.junit.tests.assertion.MultipleFailureExceptionTest$ExpectedException):java.util.List|abducedRetvar = val$2<span class='color_orange'>formal</span>:java.util.List*( sub ); return = EXN _t$0<span class='color_orange'>update:42</span>:void; errors = 0<span class='color_orange'>nullify</span>:java.util.List*; this = val$1<span class='color_orange'>formal</span>:org.junit.tests.assertion.MultipleFailureExceptionTest*( sub ); old_this = val$1<span class='color_orange'>formal</span>:org.junit.tests.assertion.MultipleFailureExceptionTest*( sub ); e = _t$3<span class='color_orange'>initial</span>:org.junit.tests.assertion.MultipleFailureExceptionTest$ExpectedException* ;
    exception|->{}<span class='color_orange'>alloc</span>:org.junit.tests.assertion.MultipleFailureExceptionTest$ExpectedException; _t$0|->{}<span class='color_orange'>alloc</span>:java.lang.AssertionError; $RET_java.util.Collections.singletonList(org.junit.tests.assertion.MultipleFailureExceptionTest$ExpectedException):java.util.List|abducedRetvar|->{mIsEmpty:1<span class='color_orange'>formal(z)</span>}<span class='color_orange'>formal(z)</span>:java.util.List( sub )</span>
    Instruction Returns
    PROP 1 of 1:
    <span class='color_blue'><span class='color_blue'>MEMne &lt; new:38 &gt; <span class='color_orange'>{vpath: }</span>(exception)</span>; <span class='color_blue'>MEMne &lt; void Assert.fail():42 &gt; <span class='color_orange'>{vpath: }</span>(_t$0)</span>; <span class='color_blue'>UND &lt; List Collections.singletonList(MultipleFailureExceptionTest$ExpectedException) &gt; :39($RET_java.util.Collections.singletonList(org.junit.tests.assertion.MultipleFailureExceptionTest$ExpectedException):java.util.List|abducedRetvar)</span> ;
    <span class='color_blue'>exception = _t$1<span class='color_orange'>update:38</span>:java.lang.Throwable*</span>; <span class='color_blue'>CatchVar25 = _t$2<span class='color_orange'>initial</span>:void</span>; <span class='color_blue'>$RET_java.util.Collections.singletonList(org.junit.tests.assertion.MultipleFailureExceptionTest$ExpectedException):java.util.List|abducedRetvar = val$2<span class='color_orange'>formal</span>:java.util.List*( sub )</span>; <span class='color_blue'>return = EXN _t$0<span class='color_orange'>update:42</span>:void</span>; <span class='color_blue'>errors = 0<span class='color_orange'>nullify</span>:java.util.List*</span>; <span class='color_blue'>this = val$1<span class='color_orange'>formal</span>:org.junit.tests.assertion.MultipleFailureExceptionTest*( sub )</span>; <span class='color_blue'>old_this = val$1<span class='color_orange'>formal</span>:org.junit.tests.assertion.MultipleFailureExceptionTest*( sub )</span>; <span class='color_blue'>e = _t$3<span class='color_orange'>initial</span>:org.junit.tests.assertion.MultipleFailureExceptionTest$ExpectedException*</span> ;
    <span class='color_blue'>exception|->{}<span class='color_orange'>alloc</span>:org.junit.tests.assertion.MultipleFailureExceptionTest$ExpectedException</span>; <span class='color_blue'>_t$0|->{}<span class='color_orange'>alloc</span>:java.lang.AssertionError</span>; <span class='color_blue'>$RET_java.util.Collections.singletonList(org.junit.tests.assertion.MultipleFailureExceptionTest$ExpectedException):java.util.List|abducedRetvar|->{mIsEmpty:1<span class='color_orange'>formal(z)</span>}<span class='color_orange'>formal(z)</span>:java.util.List( sub )</span></span>
    
    .... After Symbolic Execution ....
    PROP 1 of 1:
    <span class='color_blue'><span class='color_red'>MEMne &lt; new:38 &gt; <span class='color_orange'>{vpath: }</span>(exception)</span>; <span class='color_red'>MEMne &lt; void Assert.fail():42 &gt; <span class='color_orange'>{vpath: }</span>(_t$0)</span>; <span class='color_blue'>UND &lt; List Collections.singletonList(MultipleFailureExceptionTest$ExpectedException) &gt; :39($RET_java.util.Collections.singletonList(org.junit.tests.assertion.MultipleFailureExceptionTest$ExpectedException):java.util.List|abducedRetvar)</span> ;
    <span class='color_blue'>exception = <span class='color_red'>_t$1<span class='color_orange'>update:38</span></span>:java.lang.Throwable*</span>; <span class='color_blue'>CatchVar25 = _t$2<span class='color_orange'>initial</span>:void</span>; <span class='color_blue'>$RET_java.util.Collections.singletonList(org.junit.tests.assertion.MultipleFailureExceptionTest$ExpectedException):java.util.List|abducedRetvar = val$2<span class='color_orange'>formal</span>:java.util.List*( sub )</span>; <span class='color_blue'>return = <span class='color_red'>EXN _t$0<span class='color_orange'>update:42</span></span>:void</span>; <span class='color_blue'>errors = 0<span class='color_orange'>nullify</span>:java.util.List*</span>; <span class='color_blue'>this = val$1<span class='color_orange'>formal</span>:org.junit.tests.assertion.MultipleFailureExceptionTest*( sub )</span>; <span class='color_blue'>old_this = val$1<span class='color_orange'>formal</span>:org.junit.tests.assertion.MultipleFailureExceptionTest*( sub )</span>; <span class='color_blue'>e = <span class='color_red'>_t$3<span class='color_orange'>initial</span></span>:org.junit.tests.assertion.MultipleFailureExceptionTest$ExpectedException*</span> ;
    <span class='color_red'>exception|->{}<span class='color_orange'>alloc</span>:org.junit.tests.assertion.MultipleFailureExceptionTest$ExpectedException</span>; <span class='color_red'>_t$0|->{}<span class='color_orange'>alloc</span>:java.lang.AssertionError</span>; <span class='color_blue'>$RET_java.util.Collections.singletonList(org.junit.tests.assertion.MultipleFailureExceptionTest$ExpectedException):java.util.List|abducedRetvar|->{mIsEmpty:1<span class='color_orange'>formal(z)</span>}<span class='color_orange'>formal(z)</span>:java.util.List( sub )</span></span>
    
    
    
  </LISTING>